---
title: Where are Innovative NGS Algorithms Coming from?
tags: []
categories:
- blog
---
![](http://www.homolog.us/blogs/wp-content/uploads/2012/08/places-300x122.png)
<!--more-->

We went through our list of papers on [NGS genome
assembly](http://www.homolog.us/blogs/2012/08/07/building-a-ngs-reference-
list-please-add/) and [SNP/Variant
calling](http://www.homolog.us/blogs/2012/08/10/building-an-ngs-reference-
list-reference-genome-exists/) to find out few highly productive labs. The
list of labs is not complete and will be updated as we go through more papers.
The numbers on the map are arbitrary and do not represent any ranking system.

**Highly productive**

1\. a) [Johns Hopkins - Steven Salzberg group](http://bioinformatics.igm.jhmi.
edu/salzberg/Salzberg/Salzberg_Lab_Home.html)

b) [University of Maryland - Mihai Pop](http://www.cbcb.umd.edu/~mpop/)

2\. [Broad Institute, Boston, MA](http://www.broadinstitute.org/)

3\. [EBI, UK](http://www.ebi.ac.uk/)

4\. [BGI, Shenzen, China](http://www.genomics.cn/index)

5\. a. [Michael Smith Genome Sciences Center, Vancouver,
Canada](http://www.bcgsc.ca/platform/bioinfo/software/abyss/)

b. [University of Washington, Seattle](http://www.cs.washington.edu/~ruzzo/)

**Coming up**

6\. [JGI, Walnut Creek, CA](http://www.jgi.doe.gov/)

7\. [Michigan State University](http://ged.msu.edu/)

8\. [NTU, Singapore](http://www.ntu.edu.sg/Pages/home.aspx)

9\. [NICTA, Australia](http://www.nicta.com.au/people/tconway)

10\. [not in map yet] [GenScale, IRISA,
France](http://www.inria.fr/en/teams/genscale)

**Few observations**

You will notice that traditional universities with strong computer science or
biology departments, such as MIT, Stanford, Yale, Caltech, University of
Wisconsin, etc. are surprisingly absent from the map. Why are they not
interested in this challenging and fast-moving field that combines high-
performance computing and large-scale biology?

We believe the big US universities are so busy chasing their own tails that
they stopped solving real problems. They are well-funded and the did not face
practical problems perceived by NGS researchers.

Let us go one step back and think about the real problem faced by NGS. It is
**scaling**. Big universities usually solve scaling problems by buying large
computers or moving to supercomputers. The scaling issue is so severe for
genome centers that they could not keep throwing money at it. They had to
innovate their way out of it. So, all labs listed in 1-5 were closely
associated with genome centers, or genome centers themselves.

Why is JGI not in the top list? Once again, it is very likely that money is
their real problem. As we discussed, money and innovating to scale are
mutually exclusive.

We are seeing an interesting development. In the earlier days, biologists used
to complain endlessly about how slow the sequencing centers were in responding
to them, and dream about the number of problems they could solve, if each of
them had a mini-JGI in their labs. Now the mini-JGIs reached their labs, but
their keys to algorithms are still mostly being held by the sequencing
centers.

With that background, the achievements of 7-9, who developed scaling
algorithms without being associated with sequencing centers, is rather
remarkable. Once again, we suspect money is the driving force for their
innovations, but it is the lack of money and not rain fall of money.

