---
title: CHANCE - A Comprehensive and Easy-to-use Software for ChIP-seq QC
tags: []
categories:
- blog
---
Reader Aaron Diaz from Jun S Song's lab in UCSF alerted us about their
[recently published paper](http://genomebiology.com/2012/13/10/R98/abstract)
on new software for quality control and validation of ChIP-seq data. We never
wrote about ChIP-seq and thought this would be a good opportunity to mention
few other key papers as well.
<!--more-->

**CHANCE Download**

[Program](https://github.com/songlab/chance)

[Installation instructions](https://github.com/songlab/chance/blob/master/chan
ce_source/INSTALL)

[Tutorial](https://github.com/songlab/chance/blob/master/chance_source/TUTORIA
L)

[Manual](https://github.com/songlab/chance/blob/master/CHANCE-Manual.pdf)

**Description**

Based on description in the paper, the program developed by Aaron stands out
in two respects - (a) in its sheer number of new features compared to other
programs, (b) in its user-friendly graphical interface to make analysis easy.

From their abstract:

> CHANCE (CHip-seq ANalytics and Condence Estimation) is a standalone package
for ChIP-seq quality control and protocol optimization. Our user-friendly
graphical software quickly estimates the strength and quality of
immunoprecipitations, identies biases, compares the user's data with ENCODE's
large collection of published datasets, performs multi-sample normalization,
checks against qPCR-validated control regions, and produces informative
graphical reports.

However, the best description of CHANCE's features comes from [supplementary
CHANCE feature comparison
table](http://genomebiology.com/imedia/1633084759811870/supp1.pdf) and Fig. 8
in the paper, as shown below.

![](http://www.homolog.us/blogs/wp-
content/uploads/2012/11/Capture5-300x238.png)

![](http://www.homolog.us/blogs/wp-
content/uploads/2012/11/Capture23-300x168.png)

**Song lab at UCSF**

We also checked [the webpage of Song lab at
UCSF](http://songlab.ucsf.edu/People.html) and found it to be a collection of
brilliant people, all from theoretical physics and mathematics background.
Those interested in systems biology of transcription regulation may enjoy [the
following notes from Dr. Song's lecture](http://compbio.pbworks.com/w/page/162
52926/System%20biology%20of%20transcription%20regulation).

**Key ChIP-seq Analysis Papers**

While on the topic of ChIP-seq, we wanted to list few old and new papers
useful for data analysis.

[Genome-Wide Mapping of in Vivo Protein-DNA Interactions, David S. Johnson,
Ali Mortazavi, Richard M. Myers, Barbara Wold, Science 2007.

It was the earliest ChIP-seq paper. Following [its links at google
scholar](http://scholar.google.com/scholar?hl=en&q=Genome-
Wide+Mapping+of+in+Vivo+Protein-DNA+Interactions&btnG=&as_sdt=1%2C48&as_sdtp=)
is a good method to track the history of the field.

[Model-based Analysis of ChIP-Seq (MACS), Yong Zhang et al. Genome Biology
2008.](http://www.biomedcentral.com/content/pdf/gb-2008-9-9-r137.pdf)

It is among the most cited papers for analyzing ChIP-seq data. Shirley Liu's
group developed programs for analysis of microarray data, and this one
extended the methods.

[Computation for ChIP-seq and RNA-seq studies, Shirley Pepke, Barbara Wold &
Ali Mortazavi, Nature Methods 6, S22 - S32
(2009).](http://www.nature.com/nmeth/journal/v6/n11s/pdf/nmeth.1371.pdf)

It is a good review paper we do not like to cite for its $32 cost. We wanted
to move on to the next one, but thanks to the Chinese, [found a free copy for
you](http://www.tongji.edu.cn/~czjiang/course/paper/Review-Computation%20for
%20ChIP-seq%20and%20RNA-seq%20studies-NatMethSupp09_by_PepkeS.pdf).

[Evaluation of Algorithm Performance in ChIP-Seq Peak Detection, Elizabeth G.
Wilbanks, Marc T. Facciotti, PLOS One, 2010](http://www.plosone.org/article/in
fo%3Adoi%2F10.1371%2Fjournal.pone.0011471)

Absolutely fascinating paper. Authors not only compared all existing ChIP-seq
analysis tools available in 2010, but had enough sense to publish their
findings in PLOS One so that others can read.

**Other ChIP-seq Analysis Papers**

Choices of the following papers are arbitrary. They do not provide a full list
of all relevant papers, but only a subset. Please feel free to add in comment
section, if you think we missed a relevant and interesting paper that our
readers may enjoy.

[A signalnoise model for significance analysis of ChIP-seq with negative
control, Han Xu et al. Bioinformatics,
2010.](http://bioinformatics.oxfordjournals.org/content/26/9/1199.long)

[A co-localization model of paired ChIP-seq data using a large ENCODE data set
enables comparison of multiple samples, Kazumitsu Maehara et al. Nucl. Acids
Res. (2012).](http://nar.oxfordjournals.org/content/early/2012/11/02/nar.gks10
10.abstract)

[Picking ChIP-seq peak detectors for analyzing chromatin modification
experiments, Mariann Micsinai et al. Nucl. Acids Res.
(2012).](http://nar.oxfordjournals.org/content/40/9/e70.short)

[ChIP-Seq: technical considerations for obtaining high-quality data, Benjamin
L Kidder, Nature Immunology
(2011).](http://www.nature.com/ni/journal/v12/n10/abs/ni.2117.html)

[Detecting differential binding of transcription factors with ChIP-seq, Kun
Liang et al. Bioinformatics
(2012).](http://bioinformatics.oxfordjournals.org/content/28/1/121.abstract)

[An integrated software system for analyzing ChIP-chip and ChIP-seq data,
Hongkai Ji et al. Nature Biotechnology 26, 1293 - 1300
(2008).](http://www.nature.com/nbt/journal/v26/n11/abs/nbt.1505.html)

[PeakRanger: A cloud-enabled peak caller for ChIP-seq data, Xin Feng, Robert
Grossman and Lincoln Stein, BMC Bioinformatics
(2011).](http://www.biomedcentral.com/1471-2105/12/139/)

**Our Paper**

Finally, our modest contribution to the world of ChIP-chip and ChIP-seq:

[High resolution mapping of Twist to DNA in Drosophila embryos: Efficient
functional analysis and evolutionary conservation, Anil Ozdemir et al. Genome
Research (2011).](http://genome.cshlp.org/content/early/2011/03/04/gr.104018.1
09.abstract)

The above research was done by the group of Angelike Stathopoulos, who is a
very smart biologist at Caltech working on figuring out the transcriptional
network in _Drosophila_.

