---
title: Building an NGS Reference List (SNP&#47;variant calling category)
tags: []
categories:
- blog
---
The de novo assembly related list is
[here](http://www.homolog.us/blogs/2012/08/07/building-a-ngs-reference-list-
please-add/) and is in better shape. We are continuing to improve that one by
adding all links, some notes on each paper or section, etc.
<!--more-->

This list is for mapping, SNP, variant calling etc. A more clean distinction
is that if the paper/algorithm assumes existence of reference genome, it is
listed here. If the paper does not assume the existence of reference genome,
it goes to the other set.

**Mapping**

Langmead B, Trapnell C, Pop M, Salzberg S (2009) **Ultrafast and memory-
efficient alignment of short dna sequences to the human genome.** _Genome
Biol_ 10: R25.

Li H, Ruan J, Durbin R (2008) **Mapping short DNA sequencing reads and calling
variants using mapping quality scores.**_Genome Res._ 18(11):1851-8.
[Link](http://genome.cshlp.org/content/18/11/1851.long)

SOAP

BWT

Lunter G, Goodsib M. (2010) **Stampy: A statistical algorithm for sensitive
and fast mapping of Illumina sequence reads, **_Genome Research_.
[Link](http://genome.cshlp.org/content/early/2010/10/25/gr.111120.110)

MUMmer

**Variant Calling**

# H. Li _et al._ (2009) **The Sequence Alignment/Map format and SAMtools **

Genomic dark matter: the reliability of short read mapping illustrated by the
genome mappability score Bioinformatics (2012) 28(16): 2097-2105

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/16
/2097)

The modENCODE Data Coordination Center: lessons in harvesting comprehensive
experimental details Database (2012) 2011(0): bar023

[Abstract](http://database.oxfordjournals.org/cgi/content/abstract/2011/0/bar0
23)

A solid quality-control analysis of AB SOLiD short-read sequencing data Brief
Bioinform (2012) 0(2012): bbs048v1-bbs048

[Abstract](http://bib.oxfordjournals.org/cgi/content/abstract/bbs048v1)

MuSiC: Identifying mutational significance in cancer genomes Genome Res (2012)
22(8): 1589-1598

[Abstract](http://genome.cshlp.org/cgi/content/abstract/22/8/1589)

Gowinda: unbiased analysis of gene set enrichment for genome-wide association
studies Bioinformatics (2012) 28(15): 2084-2085

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/15
/2084)

CUSHAW: a CUDA compatible short read aligner to large genomes based on the
Burrows-Wheeler transform Bioinformatics (2012) 28(14): 1830-1837

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/14
/1830)

Strelka: accurate somatic small-variant calling from sequenced tumor-normal
sample pairs Bioinformatics (2012) 28(14): 1811-1817

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/14
/1811)

BEDOPS: high-performance genomic feature operations Bioinformatics (2012)
28(14): 1919-1920

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/14
/1919)

Fast and sensitive mapping of bisulfite-treated sequencing data Bioinformatics
(2012) 28(13): 1698-1704

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/13
/1698)

SAVoR: a server for sequencing annotation and visualization of RNA structures
Nucleic Acids Res (2012) 40(W1): W59-W64

[Abstract](http://nar.oxfordjournals.org/cgi/content/abstract/40/W1/W59)

VARIANT: Command Line, Web service and Web interface for fast and accurate
functional characterization of variants found by Next-Generation Sequencing
Nucleic Acids Res (2012) 40(W1): W54-W58

[Abstract](http://nar.oxfordjournals.org/cgi/content/abstract/40/W1/W54)

SVM2: an improved paired-end-based tool for the detection of small genomic
structural variations using high-throughput single-genome resequencing data
Nucleic Acids Res (2012) 0(2012): gks606v1-gks606

[Abstract](http://nar.oxfordjournals.org/cgi/content/abstract/gks606v1)

Dynamic hydroxymethylation of deoxyribonucleic acid marks differentiation-
associated enhancers Nucleic Acids Res (2012) 0(2012): gks595v1-gks595

[Abstract](http://nar.oxfordjournals.org/cgi/content/abstract/gks595v1)

Global Analysis of the Regulon of the Transcriptional Repressor LexA, a Key
Component of SOS Response in Mycobacterium tuberculosis J. Biol. Chem. (2012)
287(26): 22004-22014

[Abstract](http://www.jbc.org/cgi/content/abstract/287/26/22004)

Bisulfite sequencing of chromatin immunoprecipitated DNA (BisChIP-seq)
directly informs methylation status of histone-modified DNA Genome Res (2012)
22(6): 1120-1127

[Abstract](http://genome.cshlp.org/cgi/content/abstract/22/6/1120)

Taking the Next Step: Building an Arabidopsis Information Portal Plant Cell
(2012) 24(6): 2248-2256

[Abstract](http://www.plantcell.org/cgi/content/abstract/24/6/2248)

RNA-SeQC: RNA-seq metrics for quality control and process optimization
Bioinformatics (2012) 28(11): 1530-1532

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/11
/1530)

pIRS: Profile-based Illumina pair-end reads simulator Bioinformatics (2012)
28(11): 1533-1535

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/11
/1533)

A new strategy to reduce allelic bias in RNA-Seq readmapping Nucleic Acids Res
(2012) 0(2012): gks425v1-gks425

[Abstract](http://nar.oxfordjournals.org/cgi/content/abstract/gks425v1)

Sensitive mapping of recombination hotspots using sequencing-based detection
of ssDNA Genome Res (2012) 22(5): 957-965

[Abstract](http://genome.cshlp.org/cgi/content/abstract/22/5/957)

SAMSCOPE: an OpenGL-based real-time interactive scale-free SAM viewer
Bioinformatics (2012) 28(9): 1276-1277

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/9/
1276)

Integrative Genomics Viewer (IGV): high-performance genomics data
visualization and exploration Brief Bioinform (2012) 0(2012): bbs017v1-bbs017

[Abstract](http://bib.oxfordjournals.org/cgi/content/abstract/bbs017v1)

YOABS: yet other aligner of biological sequences--an efficient linearly
scaling nucleotide aligner Bioinformatics (2012) 28(8): 1070-1077

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/8/
1070)

A Fine-Scale Chimpanzee Genetic Map from Population Sequencing Science (2012)
336(6078): 193-198

[Abstract](http://www.sciencemag.org/cgi/content/abstract/336/6078/193)

Experimental Analysis of Sources of Error in Evolutionary Studies Based on
Roche/454 Pyrosequencing of Viral Genomes Genome Biol Evol (2012) 4(4):
457-465

[Abstract](http://gbe.oxfordjournals.org/cgi/content/abstract/4/4/457)

Detecting Selective Sweeps from Pooled Next-Generation Sequencing Samples Mol
Biol Evol (2012) 0(2012): mss090v2-mss090

[Abstract](http://mbe.oxfordjournals.org/cgi/content/abstract/mss090v2)

JointSNVMix: a probabilistic model for accurate detection of somatic mutations
in normal/tumour paired next-generation sequencing data Bioinformatics (2012)
28(7): 907-913

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/7/
907)

Breakpointer: using local mapping artifacts to support sequence breakpoint
discovery from single-end reads Bioinformatics (2012) 28(7): 1024-1025

[Abstract](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/28/7/
1024)

Genotype calling from next-generation sequencing data using haplotype
information of reads Bioinformatics (2012) 28(7): 938-946 [Abstract](http://bi
oinformatics.oxfordjournals.org/cgi/content/abstract/28/7/938)

Analysis of insertion-deletion from deep-sequencing data: software evaluation
for optimal detection Brief Bioinform (2012) 0(2012): bbs013v1-bbs013
[Abstract](http://bib.oxfordjournals.org/cgi/content/abstract/bbs013v1)

