---
title: Humor - Announcement of a New Sequencing Technology
tags: []
categories:
- blog
---
For today's comic relief, here is a press release forwarded to us by a reader.
<!--more-->

>

**A new sequencing technology enters the ring: SHTseq(TM)**

Just received this press release from Neil Hall, CEO of a new sequencing
company called CrapBio whos launching at this years #notAGBT.. sounds very
interesting!

**CrapBio**

CrapBio is excited to announce the release of its new SHTseq whole genome
sequencing platform which is going to revolutionize the genomics industry and
soon the healthcare industry.

**SHTseq**

SHTseq or Super High-throughput Template sequencing is a totally new paradigm
for DNA analysis. In the era of second and third generation sequencing,
scientists would take days to extract DNA and prepare it for analysis on slow
cumbersome analysis machines. CrapBio is able to get around this using
AngstromRealtimeSensors that can accurately read the DNA while still inside
cells. Better than that, you dont even need to take the DNA to the analysis
machine you can simply take an image of the organism you wish to sequence and
upload it to the SHTcloud and have the DNA information extracted directly by
out highly trained SHTtechnicians.

With SHTseq you can simply send the sample from you iphone and have the
sequencing data returned to your email address in a matter of minutes. THATS
RIGHT!, you can have a SHT experiment run in minutes! Not hours or days.
Download your iSHT app now and start sequencing.

Longer Reads-Better Data noSHT (What would you do with 100Mb reads?).

We are able to generate super-long reads with our ARSesnsors. Using CrapBio-
SHTseq technology we regularly get 10Mb reads and we have even seen reads of
100Mb which completely sequenced E. coli 20 times in a single read. Our base
calling accuracy is 25%, but with genomes with extreme AT/GC bias it reaches
40%. Although this is lower than other platforms the longer reads allow you to
extract much more information from our reads than old-fashioned 2nd generation
sequencers. Also this error is totally randomly distributed (unlike
homopolymer errors in other technologies!) and there is no decline in base
calling accuracy toward the ends of reads. The last base in a read is just as
good as the first base.

Please follow the rest of the press release [at this
link](http://pathogenomics.bham.ac.uk/blog/2012/02/a-new-sequencing-
technology-enters-the-ring-shtseqtm/).

\-----------------------

And here is something for the 'not so humorous' column. We just found out that
the website hosting the largest number of software codes to analyze ABI SOLiD
data has been abandoned.

![](http://www.homolog.us/blogs/wp-content/uploads/2012/06/Capture-300x90.png)

Anyone knows what is going on? What has the world come to during our month of
absence?

