---
title: "#bog13 Posters on Pine and Coelacanth"
tags: []
categories:
- blog
---
We decided to post on talks and posters from Biology of Genome 2013 meeting
for our readers, who could not attend the meeting at CSHL. The process has
been quite time consuming, because neither Cold Spring Harbor Lab nor
conference organizers posted abstracts and detailed list of authors online.
So, we have to Google the first name of only author, find his email address
and then request him personally to get a copy of her slides or posters.
<!--more-->

One request was rejected with the excuse that our blog is not 'neutral'. The
snub is likely related to our reporting of ENCODE hype. The characterization
is odd, because we try to be as neutral as science allows us to be, and never
distorted a scientific analysis that is convincingly correct. Unfortunately
science itself is quite brutal to bad theories. In fact, science can be
described as extremely biased by contemporary media (think [CNN
Crossfire](http://en.wikipedia.org/wiki/Crossfire_\(TV_series\))), because it
does not place an evolution pundit and and intelligent design pundit on the
same podium and let them 'debate' on the issues.

That aside, we received few slides and posters and will continue to post them
in the coming days.

**Loblolly Pine**

It is quite an odd day to keep switching between longest sequenced plant
genome (22 Gb) and smallest one (82Mb). Earlier we wrote about Steven
Salzberg's talk on [genome assembly of loblolly
pine](http://www.homolog.us/blogs/2013/05/11/steven-salzberg-at-bog13
-assembling-22gb-conifer-genome/). Daniela Puiu from John Hopkins university
sent us the associated poster that our readers may find informative.

In it, their group presented some comparative statistics after assembling
their reads with SOAPdenovo2 assembler. We have one complaint about their
conclusion however. Based on our understanding of the algorithm, short
SOAPdenovo2 contig is a feature and not a bug. If we understand correctly,
SOAPdenovo2 makes up in their scaffolding step by proceeding hierarchically
from PE reads to reads with longer mate pairs.

Please click on the image to see a full view of their poster.

![Capture1](http://www.homolog.us/blogs/wp-
content/uploads/2013/05/Capture18-300x224.png)

Regarding availability of their program, Dr. Salzberg and Daniela Puiu
informed us:

DP:

> The MaSuRCA assembler is available on the University of Maryland ftp server
under: ftp://ftp.genome.umd.edu/pub/MSR-CA/

SS:

> The specific pipeline we're using for pine isn't in any format we could
share right now, but as Daniela says, the code is available. There's lots of
other pieces too - all will be made freely available as soon as we can.

**Coelacanth**

We earlier wrote about [Coelacanth genome
paper](http://www.homolog.us/blogs/2013/04/19/genome-of-a-living-fossil-2/)
and [their google hangout meeting](http://www.homolog.us/blogs/2013/04/23
/google-hangout-meeting-on-coelacanth-genome-paper-and-controversy/). Chris
Amemiya sent us his BOG13 slides that our readers may find informative.

When you click on the link, you will download the zip file and will have to
unzip it in your computer. The poster is in Powerpoint format.

![Capture2](http://www.homolog.us/blogs/wp-
content/uploads/2013/05/Capture24-300x300.png)

