---
title: How Real Are Those 600,000 Spliced Genes in Your Transcriptome Data?
tags: []
categories:
- rnaseq
---
Those working on RNAseq assembly and analysis may find [this 2010 paper](http:
//www.ncbi.nlm.nih.gov/pmc/articles/PMC3000347/pdf/pgen.1001236.pdf) helpful.
(h/t: @DanGraur)
<!--more-->

[Noisy Splicing Drives mRNA Isoform Diversity in Human Cells](http://www.ncbi.
nlm.nih.gov/pmc/articles/PMC3000347/pdf/pgen.1001236.pdf)

> While the majority of multiexonic human genes show some evidence of
alternative splicing, it is unclear what fraction of observed splice forms is
functionally relevant. In this study, we examine the extent of alternative
splicing in human cells using deep RNA sequencing and de novo identification
of splice junctions. We demonstrate the existence of a large class of low
abundance isoforms, encompassing approximately 150,000 previously unannotated
splice junctions in our data. Newly identified splice sites show little
evidence of evolutionary conservation, suggesting that the majority are due to
erroneous splice site choice. We show that sequence motifs involved in the
recognition of exons are enriched in the vicinity of unconserved splice sites.
We estimate that the average intron has a splicing error rate of approximately
0.7% and show that introns in highly expressed genes are spliced more
accurately, likely due to their shorter length. These results implicate noisy
splicing as an important property of genome evolution.

