---
title: Large-scale Mouse Deletion Study Published
tags: []
categories:
- blog
---
Among a number of large-scale experiments reported on yeast between 1999-2005
(gene expression, protein-protein interaction, transcription factor binding,
etc.), yeast deletion papers by Amy Hin Yan Tong et al. ([2001](http://rna.gen
omics.purdue.edu/@api/deki/files/1185/=Systematic_Genetic_Analysis_with_Ordere
d_Arrays_of_Yeast_Deletion_Mutants.pdf) and [2004](http://angel.elte.hu/fij
/old-homepage/okt/gbi/kieg/pdf/Tong-04-SCI-
Global%20Mapping%20of%20the%20Yeast%20Genetic%20Interaction%20Network.pdf))
got particular attention, because they connected genes with phenotypes. It is
easy to argue that a transcription factor binding measurement is incorrect or
irrelevant, but it is too hard to argue against an essential gene.
<!--more-->

Now Jacqueline K. White and colleagues at Sanger Institute Mouse Genetics
Project make an equally important large-scale knock out study on mice publicly
available (h/t: @dgmacarthur). The paper is full of useful insights, among
which the finding that novel essential genes are as likely to be related to
disease as well-studied ones is the most striking. It will be quite
fascinating, when this project gets to cover the entire set of genes in mouse.
The paper is open-access and the mice are getting proper care in accordance
with the UK Home Of?ce regulations. So, you have very little to complain
about.

[Genome-wide Generation and Systematic Phenotyping of Knockout Mice Reveals
New Roles for Many Genes](http://download.cell.com/pdf/PIIS0092867413007617.pd
f?intermediate=true)

> Mutations in whole organisms are powerful ways of interrogating gene
function in a realistic context. We describe a program, the Sanger Institute
Mouse Genetics Project, that provides a step toward the aim of knocking out
all genes and screening each line for a broad range of traits. We found that
hitherto unpublished genes were as likely to reveal phenotypes as known genes,
suggesting that novel genes represent a rich resource for investigating the
molecular basis of disease. We found many unexpected phenotypes detected only
because we screened for them, emphasizing the value of screening all mutants
for a wide range of traits. Haploinsuf?ciency and pleiotropy were both
surprisingly common. Forty-two percent of genes were essential for viability,
and these were less likely to have a paralog and more likely to contribute to
a protein complex than other genes. Phenotypic data and more than 900 mutants
are openly available for further analysis.

