---
title: Google Hangout Meeting on Coelacanth Genome Paper (and 'Controversy')
tags: []
categories:
- blog
---
[A most amazing fish: Join our Google Hangout about coelacanths on
Thursday](http://phenomena.nationalgeographic.com/2013/04/22/a-most-amazing-
fish-join-our-google-hangout-about-coelacanths-on-thursday/)
<!--more-->

Carl Zimmer at National Geographic organizes a Google hangout meeting with the
authors of coelacanth genome paper -

>

Which is why I hope youll join me Thursday at 11 am ET to participate in a
Google Hangout with a panel of scientists to talk about the latest scientific
discoveries about this amazing fish. The occasion is the publication of the
coelacanth genome last week.

Now, if you read the Loom with any regularity, you know that the mere
publication of a genome is not, in my opinion, automatically news. But the
scientists who sequenced the coelacanth genome have analyzed it to explore
some very interesting questions, and their work has provided some intriguing
clues about the evolution of our limbs and many other aspects of our biologyas
well as some puzzling features unique to coelacanths themselves. The genome
paper has also inspired some criticism in the blogosphere.

What kind of criticism? Please check the Casane and Laurenti (2013) review
paper mentioned by Dave Lunt in [our previous
commentary](http://www.homolog.us/blogs/2013/04/19/genome-of-a-living-fossil-2
/comment-page-1/). After receiving the comment, we contacted Dr. Laurenti
asking whether the genome paper satisfied his objection. He agreed that "Chris
Amemiya's staff did an impressive work in showing that the substitution rate
was about 25% inferior compared to the one of tetrapods". However, his
criticism was about whether 'living fossil' description of coelacanth was
correct. Did the morphology of 'Latimeria' stay static over 300 million years?
That assertion is not backed by fossil record published in paleontological
literature.

[In this humorous blog
commentary](http://freethoughtblogs.com/pharyngula/2013/04/20/coelacanths-are-
unexceptional-products-of-evolution/comment-
page-1/?utm_source=feedburner&utm_medium=feed), P Z Myers 'translated' Casane
and Laurenti paper from journal-speak to blog-speak (or should we say Dan-
Graur-speak?),

>

Snorble-garble-ptang-ptang-CLUNK. Reset. Does not compute. Must recalibrate
brain.

None of that makes sense. The modern fish do not show a striking resemblance
to their fossilize ancestors they retain skeletal elements that link them to a
clade thought to be extinct. This assumption that Actinistian infraclass has
been unchanging undermines their conclusions the modern species are different
enough that theyve been placed in a unique genus not shared with any fossil
form.

Then the argument that they must live in a stable environment with a lack of
natural-selection pressure is absurd. Selection is generally a conservative
process: removing selection pressures from a population should lead to an
increase in the accumulation of variability. Do they mean there has been
increased selection in a very narrowly delimited but stable environment?

But even that makes no sense. We should still be seeing the accumulation of
neutral alleles. Increased selection is only going to remove variability in
functional elements, and most of the genome isnt. I suppose one alternative to
explain slow molecular evolution would be extremely high fidelity replication,
but even that would require specific selection constraints to evolve.

and posted an interesting observation -

> The authors of the BioEssays article made an interesting, and troubling
analysis: it depends on the authors theoretical priors. They examined 12
relevant papers on coelacanth genes published since 2010, and discovered a
correlation: if the paper uncritically assumed the living fossil hypothesis
(which Ive told you is bunk), the results in 4 out of 5 cases concluded that
the genome was slowly evolving; in 7 out of 7 cases in which the work was
critical of the living fossil hypothesis or did not even acknowledge it, they
found that coelacanth genes were evolving at a perfectly ordinary rate.

Is this heading to an ENCODE-like showdown? Hopefully, the authors of
coelacanth genome paper will properly address this issue in the google hangout
discussion.

