---
title: Methods to study Event&#47;Isoform Expression and Alternative Splicing from
  RNA-Seq
tags: []
categories:
- rnaseq
---
[RegulatoryGenomics website](http://regulatorygenomics.upf.edu/Software/RNA-
Seq_and_splicing/) posts and updates a comprehensive list of tools for RNAseq
analysis, and the current version is posted below. If you are overwhelmed with
too many tools, please wait till our next commentary, where we will put them
all in context. Also check [this seqanswers
thread](http://seqanswers.com/forums/showthread.php?p=98949#post98949) for
more on the topic.
<!--more-->

(please send

comments/suggestions to eduardo.eyras@upf.edu)

Spliced-mappers

Method

Reference

Web-site

TopHat

(Trapnell et al. 2009)

<http://tophat.cbcb.umd.edu/>

MapSplice

(Wang et al. 2010)

<http://www.netlab.uky.edu/p/bioinfo/MapSplice>

SpliceMap

(Auger et al. 2010)

<http://www.stanford.edu/group/wonglab/SpliceMap/>

HMMSplicer

(Dimon et al. 2010)

<http://derisilab.ucsf.edu/index.php?software=105>

TrueSight

(Li et al. 2012b)

<http://bioen-compbio.bioen.illinois.edu/TrueSight/>

SOAPsplice

(Huang et al. 2011)

<http://soap.genomics.org.cn/soapsplice.html>

PASSion

(Zhang et al. 2012)

<https://trac.nbic.nl/passion>

PALMapper

(Jean et al. 2010)

<http://galaxy.raetschlab.org/>

SplitSeek

(Ameur et al. 2010)

<http://solidsoftwaretools.com/gf/project/splitseek>

Supersplat

(Bryant et al. 2010)

<http://mocklerlab-tools.cgrb.oregonstate.edu/>

SeqSaw

(Wang et al. 2011)

http://bioinfo.au.tsinghua.edu.cn/software/seqsaw

MapNext

(Bao et al. 2009)

<http://evolution.sysu.edu.cn/english/software/mapnext.htm>

STAR

(Dobin et al. 2012)

<http://gingeraslab.cshl.edu/STAR/>

GSNAP

(Wu et al. 2010)

<http://research-pub.gene.com/gmap/>

QPALMA

(De Bona et al. 2008)

<http://www.raetschlab.org/suppl/qpalma>

OSA

(Hu et al. 2012)

<http://omicsoft.com/osa/>

De novo

Transcript assembly methods

Method

Reference

Web-site

Rnnotator

(Martin et al. 2010)

[Email to

vtdelapuente@lbl.gov

Multiple-k/STM

(Surget-Groba et al. 2010)

<http://www.surget-groba.ch/downloads/stm.tar.gz>

OASES

(Schultz et al. 2012)

<http://www.ebi.ac.uk/~zerbino/oases/>

Abyss

(Birol et al. 2009)

<http://www.bcgsc.ca/platform/bioinfo/software/abyss>

TransAbyss

(Robertson et al. 2010)

<http://www.bcgsc.ca/platform/bioinfo/software/>

Trinity

(Grabherr et al. 2011)

[http://TrinityRNASeq.sourceforge.net](http://trinityrnaseq.sourceforge.net/)

OASES

(Schultz et al. 2012)

<http://www.ebi.ac.uk/~zerbino/oases/>

Kissplice

(Sacomoto et al. 2012)

<http://alcovna.genouest.org/kissplice/>

Genome-based methods not using

annotation

Method

Reference

Web-site

G-Mo.R-Seq

(Denoeud et al. 2008)

<http://www.genoscope.cns.fr/externe/gmorse/>

TAU

(Filichkin et al. 2010)

<http://mocklerlab-tools.cgrb.oregonstate.edu/>

Scripture

(Guttman et al. 2010)

<http://www.broadinstitute.org/software/Scripture/>

Cufflinks

(Trapnell et al. 2010)

http://cufflinks.cbcb.umd.edu/

IsoLasso

(Li et al. 2011a)

<http://www.cs.ucr.edu/~liw/isolasso.html>

CEM

(Li et al. 2012a)

<http://www.cs.ucr.edu/~liw/cem.html>

NSMAP

(Xia et al. 2011)

<https://sites.google.com/site/nsmapforrnaseq>

Montebello

(Hiller et al. 2012)

<http://www.stanford.edu/group/wonglab/Montebello/Montebello_0.8.tar.gz>

Casper

(Rosell et al. 2012)

<https://sites.google.com/site/rosselldavid/software>

**Quantification of events using 

gene annotations**

Method

Reference

Web-site

RUM

(Grant et al. 2011)

<http://www.cbil.upenn.edu/RUM/userguide.php>

X-MATE

(Wood et al. 2011)

<http://grimmond.imb.uq.edu.au/X-MATE/>

MMES

(Wang et al. 2010b)

Email to Wang.Liguo@mayo.edu

SpliceTrap

(Wu et al. 2011)

<http://rulai.cshl.edu/splicetrap/>

Genome-based quantification of

isoforms using gene annotations

Method

Reference

Web-site

rSeq

(Jiang et al. 2009)

<http://www-personal.umich.edu/~jianghui/rseq/>

rQuant

(Bohnert et al. 2009)

<http://galaxy.raetschlab.org/>

SOLAS

(Richard et al. 2010)

<http://cmb.molgen.mpg.de/2ndGenerationSequencing/Solas/>

MISO

(Katz et al. 2010)

<http://genes.mit.edu/burgelab/miso/>

FluxCapacitor

(Montgomery et al. 2010)

<http://flux.sammeth.net/capacitor.html>

IsoformEx

(Kim et al. 2011)

<http://bioinformatics.wistar.upenn.edu/isoformex>

IsoInfer

(Feng et al. 2012)

<http://www.cs.ucr.edu/~jianxing/IsoInfer.html>

iReckon

(Mezlini et al. 2012)

<http://compbio.cs.toronto.edu/ireckon/>

IQSeq

(Du et al. 2012)

<http://archive.gersteinlab.org/proj/rnaseq/IQSeq/>

RABT

(Roberts et al. 2011)

[http://bio.math.berkeley.edu/cufflinks.](http://bio.math.berkeley.edu/cufflin
ks)

SLIDE

(Li et al. 2011b)

<https://sites.google.com/site/jingyijli/SLIDE.zip>

Isoform quantification using

transcriptome annotations

Method

Reference

Web-site

RSEM

(Li et al. 2011c)

<https://github.com/bli25wisc/RSEM/>

IsoEM

(Nicolae et al. 2011)

<http://dna.engr.uconn.edu/?page_id=105>

NEUMA

(Lee et al. 2011)

[http://neuma.kobic.re.kr.

BitSeq

(Glaus et al. 2012)

<http://www.bioconductor.org/packages/2.11/bioc/html/BitSeq.html>

MMSEQ

(Turro et al. 2011)

<http://bgx.org.uk/software/mmseq.html>

eXpress

(Roberts et al. 2013)

<http://bio.math.berkeley.edu/eXpress/>

Annotation-guided reconstruction

(and quantification) of isoforms

Method

Reference

Web-site

MapAI

(Labaj et al. 2012)

<http://www.bioinf.boku.ac.at/pub/MapAl/>

SpliceGrapher

(Rogers et al. 2012)

http://SpliceGrapher.sf.net

RABT

(Roberts et al. 2011)

[http://bio.math.berkeley.edu/cufflinks.](http://bio.math.berkeley.edu/cufflin
ks)

ExonMap/ JunctionWalk

(Seok et al. 2012)

NA

DRUT

(Mangul et al. 2012)

<http://www.cs.gsu.edu/~serghei/?q=drut>

SLIDE

(Li et al. 2011b)

https://sites.google.com/site/jingyijli/SLIDE.zip

Gaze

(Howe et al. 2002)

<http://www.sanger.ac.uk/resources/software/gaze/>

JigSaw

(Allen et al. 2005)

<http://www.cbcb.umd.edu/software/jigsaw/>

EVM

(Haas et al. 2008)

<http://evidencemodeler.sourceforge.net/>

Augustus

(Stanke et al. 2006a)

<http://bioinf.uni-greifswald.de/augustus/>

Evigan

(Liu et al. 2008)

[http://www.seas.upenn.edu/~strctlrn/evigan/evigan.html](http://www.seas.upenn
.edu/%7Estrctlrn/evigan/evigan.html)

Event-based differential

splicing

Method

Reference

Web-site

Alexa-Seq

(Griffith et al. 2010)

<http://www.alexaplatform.org/alexa_seq/>

MISO

(Katz et al. 2010)

<http://genes.mit.edu/burgelab/miso/>

DEXSeq

(Anders et al. 2012)

<http://www.bioconductor.org/packages/release/bioc/html/DEXSeq.html>

MATS

(Shen et al. 2012)

<http://rnaseq-mats.sourceforge.net/>

DiffSplice

(Hu et al. 2012)

http://www.netlab.uky.edu/p/bioinfo/DiffSplice

SOLAS

(Richard et al. 2010)

<http://cmb.molgen.mpg.de/2ndGenerationSequencing/Solas/>

JuncBASE

(Brooks et al. 2011)

<http://compbio.berkeley.edu/proj/juncbase/Home.html>

JETTA

(Seok et al. 2012b)

<http://igenomed.stanford.edu/~junhee/JETTA/rnaseq.html>

SpliceSeq

(Ryan et al. 2012)

<http://bioinformatics.mdanderson.org/main/SpliceSeq:Overview>

Isoform-based differential

splicing

Method

Reference

Web-site

BASIS

(Zheng et al. 2009)

