---
title: 'New AJHG Paper: ''Born Again'' Dinosaurs Question Effectiveness of #GWAS'
tags: []
categories:
- blog
---
This paper could not have come at a worst time for BioMickWatson, who was
about to classify us as dinosaurs for asking questions about the fast-rising
(and soon to be obsolete) religion of Genome-wide Association Studies (GWAS)
to heal complex diseases. Our previous discussions on the topic are linked
below -
<!--more-->

[Study History and Read Papers Written by Dinosaurs
(#GWAS)](http://www.homolog.us/blogs/blog/2013/08/12/study-history-and-read-
old-papers/)

[Genome Wide Association Studies (#GWAS) Are They
Replicable?](http://www.homolog.us/blogs/blog/2013/07/26/genome-wide-
association-studies-gwas-are-they-replicable/)

[Failure to Fund #GWAS Leads to Increase in Cancer in a Chinese
Village](http://www.homolog.us/blogs/blog/2013/07/26/in-need-for-cancer-gwas/)

[Battle over #GWAS: Ken Weiss
Edition](http://www.homolog.us/blogs/blog/2013/07/30/battle-over-gwas-ken-
weiss-edition/)

[Cancer, Stroke and Diabetes in USA Environment versus
Genetics](http://www.homolog.us/blogs/blog/2013/07/28/cancer-and-diabetes-in-
usa-environment-versus-genetics/)

The new paper, [published in American Journal of Human Genetics](http://downlo
ad.cell.com/AJHG/pdf/PIIS0002929713002826.pdf?intermediate=true) (thanks
@ProfBootyPhD for correction), investigated the validity of associations on a
disease/disability that is very relevant for this intellectual debate.

> Intellectual disability (ID, formerly called mental retardation) is a
developmental brain disorder commonly de?ned by an IQ below 70 and limitations
in both intellectual functioning and adaptive behavior. ID can originate from
environmental causes and/or genetic anomalies, and its incidence in children
is estimated to be of 1%2%.

The paper noticed something odd -

> One of the ?rst genes identi?ed as involved in XLID is FMR1 (MIM 309550), a
target of the unstable expansion mutation responsible for fragile X syndrome
(MIM 300624); accounting for about 1%2% of all ID cases, this mutation still
remains the most common cause of XLID.5,6 Since then, **the number of genes
involved in XLID when mutated has grown exponentially**, from only 11 in 1992
to 43 in 2002 and over 100 genes now identi?ed thank to the efforts of various
teams.

What if that exponential rise is due to government pouring increasing amount
of money on #GWAS? The paper by Piton et al. suggests that could be the case.
They further checked 28 genes #GWASsociated with XLID and brought down the
list to 3 'likely' genes and a large number of 'questionable' and 'highly
questionable' ones.

> We used this NHLBI cohort to systematically reassess the implication of 106
genes proposed to be involved in monogenic forms of XLID. We particularly
question the implication in XLID of ten of them (AGTR2, MAGT1, ZNF674, SRPX2,
ATP6AP2, ARHGEF6, NXF5, ZCCHC12, ZNF41, and ZNF81), in which truncating
variants or previously published mutations are observed at a relatively high
frequency within this cohort. We also highlight 15 other genes (CCDC22, CLIC2,
CNKSR2, FRMPD4, HCFC1, IGBP1, KIAA2022, KLF8, MAOA, NAA10, NLGN3, RPL10,
SHROOM4, ZDHHC15, and ZNF261) for which replication studies are warranted. We
propose that similar reassessment of reported mutations (and genes) with the
use of data from large-scale human exome sequencing would be relevant for a
wide range of other genetic diseases.

Oh well, it feels like a rerun of Jurassic park movie, when dinosaurs landed
in San Diego :)

\--------------------------------

We thank the blog of Ken Weiss for bringing this paper to our attention.
Readers will enjoy their discussion -

[Genes 'for' XLID -- the replication/falsifiability
fallacy](http://ecodevoevo.blogspot.com/2013/08/genes-for-xlid-
replicationfalsifiabilit.html)

It includes a quote from Aravinda Chakraborty, a leading human geneticist,
which starts with -

> But while finding and localizing genes isnt a big challenge any more, it is
still very difficult to figure out what they actually do.

Who could have known that in this new era of hypothesis-free and consensus-
driven 'science'?

Readers may note that ecodevoevo blog mentioned another previous study of
similar nature -

> Another cautionary tale appeared in Cell not long ago. A paper called "Exome
Sequencing of Ion Channel Genes Reveals Complex Profiles Confounding Personal
Risk Assessment in Epilepsy", Klassen et al., from Jeffrey Noebels' lab,
reported the results of sequencing the exomes of 237 ion channel genes in
people with and without sporadic idiopathic epilepsy. They found that "rare
missense variation in known Mendelian disease genes is prevalent in both
groups at similar complexity, revealing that even deleterious ion channel
mutations confer uncertain risk to an individual depending on the other
variants with which they are combined." That is, the same ion channel variants
assumed to cause epilepsy are found in people without the trait.

\---------------------------------

Edit.

BioMickWatson responded to our disagreement in an elegant way -

![Capture](http://www.homolog.us/blogs/wp-
content/uploads/2013/08/Capture39-300x57.png)

'English' of this Englishman, we presume, refers to [his response
here](http://www.homolog.us/blogs/blog/2013/08/12/study-history-and-read-old-
papers/), where he found out few trivial spelling errors in Dan Graur's paper
and compared it with incomprehensibility of layman's abstract of an eQTL
paper.

