---
title: Two Important Papers Published - Assemblathon and First Genome using Moleculo
tags: []
categories:
- blog
---
Let us clarify at the outset that today's 'important' in the title is not
being used to describe the papers themselves, but rather where they were
published. It shows a change in trend from the Nature-Science-Cell etc.
mentality to go for the new, open-source and innovative journals to publish
unusual results.
<!--more-->

The [Assemblathon 2 paper came out in
GigaScience](http://www.gigasciencejournal.com/content/2/1/10/prepub), a new
free and open-access journal backed by BGI and Springer (BMC). The best part
about the journal is its open publication of peer reviews with names of
reviewers ([Mick Watson's review](http://www.gigasciencejournal.com/imedia/790
4535709171739_comment.pdf), [C. Titus Brown's review](http://www.gigasciencejo
urnal.com/imedia/2602220779275382_comment.pdf)). That makes the process very
transparent, and places the reviewers and authors at the same level instead of
judge and defendant, who is guilty until proven otherwise. Another innovative
aspect of Gigascience journal is its GigaDb database for storing large data
sets with doi identifiers, etc.

The Assemblathon project was also run in a very next-generation friendly way,
'next-generation' of communication that is. Keith Bradnam, the first author of
the paper, [maintained a blog](http://assemblathon.org/post/56126786649/the-
assemblathon-2-paper-has-been-published) and provided daily updates on genome
assembly-related news from his @assemblathon twitter feed. He even went to the
extent of [maintaining a list with everyone's
comments](http://assemblathon.org/post/44431915644/feedback-and-analysis-of-
the-assemblathon-2-p) about the paper in his blog. How did the others manage
to comment on the paper since January, if it is published today? Well, for
anyone who cared, the paper was 'published' at the arxiv server in January.
That approach surely takes the bang away from the ultimate publication
surprise for the journals, but makes the life of the authors lot easier. We
have been trying to convince our older collaborators about how the culture is
changing, because they are still trying to maintain the old mode of 'its all
secret until the paper is in print' - a very stressful way to do science.

Also check GigaScience editor-in-chief on -

[Assemblathon 2: Unusual meta peer
review](http://www.biomedcentral.com/biome/assemblathon-2-unusual-meta-peer-
review/)

and Mr. Assemblathon on -

[Writing the Assemblathon 2 paper, an experiment in
openness](http://assemblathon.org/post/56217669322/writing-the-assemblathon-2
-paper-an-experiment-in)

[Assemblathon mailing list](https://lists.ucdavis.edu/sympa)

\----------------------------------------------------------

The other paper came out in e-life, a new free and open-access journal backed
by HHMI, Welcome trust and Max Planck Institute. Those private funding
agencies realized that the were spending too much money on publication cost,
even though the papers were being promptly locked into subscription-only
websites and they came up with their own journal. It is refreshing to find
that major papers are accepting such journals instead of going through the
same 'Nature-Science-Cell' etc. route.

[The genome sequence of the colonial chordate, Botryllus
schlosseri](http://elife.elifesciences.org/content/2/e00569)

We will discuss the Moleculo technology in a later commentary, but here is the
web-page version of the abstract.

> The tunicates are an evolutionary group that includes species such as sea
squirts and sea tulips. Their name comes from the structure known as a tunic
that surrounds their sac-like bodies. As marine filter feeders, tunicates
obtain nutrients by straining food particles from water, and they can live
either alone or in colonies depending on the species. Charles Darwin suggested
that tunicates may be the key to understanding the evolution of vertebrates,
and indeed today they are regarded as the closest living relatives of this
group.

Colonial tunicates can reproduce either sexually, or asexually by budding.
Compatible colonies have the ability to recognize one another and to fuse
their blood vessels to form a single organism, whereas incompatible colonies
reject one another and remain separate. This recognition process bears some
resemblance to the rejection of foreign organ transplants in mammals.

Here, Voskoboynik and co-workers present the first genome sequence of a
colonial tunicate, Botryllus schlosseri. They used a novel sequencing approach
that significantly increased the length of a DNA molecule that can be
determined by next-generation sequencing, and allowed large DNA repeat regions
to be easily resolved. In total, they sequenced 580 million base pairs of DNA,
which they estimate contains roughly 27,000 genes.

By comparing the B. schlosseri genome with those of a number of vertebrates,
Voskoboynik et al. identified multiple B. schlosseri genes that also
participate in the development and functioning of the vertebrate eye, heart,
and auditory system, as well as others that may have contributed to the
evolution of the immune system and of blood cells. The genome of B. schlosseri
thus provides an important new tool for studying the genetic basis of the
evolution of vertebrates.

One major problem of e-life portal is its hiring of a Javascript expert, who
never went through [the annoying.js
rules](http://www.homolog.us/blogs/software/2013/06/06/javascript-checker-
jsfiddle/). The website blocks copy-and-paste action, and we had to parse
through the html source code to display the title of the paper. Hopefully,
that will be fixed in the near future.

