---
title: 'antiSMASH: Rapid Identification, Annotation and Analysis of Secondary Metabolite
  Biosynthesis Gene Clusters'
categories:
- chem
---
We are looking at an interesting paper from M. H. Medema _et al._ on finding
gene clusters in bacterial genomes.
<!--more-->

[antiSMASH: rapid identification, annotation and analysis of secondary
metabolite biosynthesis gene clusters in bacterial and fungal genome sequences
](http://nar.oxfordjournals.org/content/early/2011/06/14/nar.gkr466.abstract)

> Bacterial and fungal secondary metabolism is a rich source of novel
bioactive compounds with potential pharmaceutical applications as antibiotics,
anti-tumor drugs or cholesterol-lowering drugs. To find new drug candidates,
microbiologists are increasingly relying on sequencing genomes of a wide
variety of microbes. However, rapidly and reliably pinpointing all the
potential gene clusters for secondary metabolites in dozens of newly sequenced
genomes has been extremely challenging, due to their biochemical
heterogeneity, the presence of unknown enzymes and the dispersed nature of the
necessary specialized bioinformatics tools and resources. Here, we present
antiSMASH (antibiotics & Secondary Metabolite Analysis Shell), the first
comprehensive pipeline capable of identifying biosynthetic loci covering the
whole range of known secondary metabolite compound classes (polyketides, non-
ribosomal peptides, terpenes, aminoglycosides, aminocoumarins,
indolocarbazoles, lantibiotics, bacteriocins, nucleosides, beta-lactams,
butyrolactones, siderophores, melanins and others). It aligns the identified
regions at the gene cluster level to their nearest relatives from a database
containing all other known gene clusters, and integrates or cross-links all
previously available secondary-metabolite specific gene analysis methods in
one interactive view.

The software is available here -

[Link](http://antismash.secondarymetabolites.org/)

The pipeline for the program is fairly simple. From the [NAR paper](http://nar
.oxfordjournals.org/content/early/2011/06/14/nar.gkr466/F1.expansion.html) \-

![Capture](http://www.homolog.us/blogs/chem/wp-content/uploads/sites/4/2013/01
/Capture-300x190.png)

Why is it important? It is because the above work and similar research
connects bioinformatics of next-gen sequencing (especially metagenomics) [to
drug discovery](http://www.xconomy.com/boston/2012/01/10/warp-drive-bio-
launches-with-125m-from-third-rock-greylock-sanofi/). We will add more on the
topic in future commentaries.

