---
title: Commercial Value of Efficient Metagenomics Pipeline
tags: []
categories:
- blog
---
[In his blog](http://ivory.idyll.org/blog/crowdsourcing-my-
research.html#disqus_thread), Titus Brown asked for ideas to make his open-
source algorithm discovery project more exciting. Here is one.
<!--more-->

Even though it appears that assembling metagenomes from ocean water, and soil
samples has little commercial value, that is not correct [according to a
Boston-based startup](http://www.xconomy.com/boston/2012/01/10/warp-drive-bio-
launches-with-125m-from-third-rock-greylock-sanofi/?single_page=true). Ideally
we should not call it a startup, because Sanofi already made commitment to buy
the company provided they meet their goals.

>

Warp Drive is being launched with $125 million in funding from Third Rock and
French pharmaceutical giant Sanofi (NYSE: SNY). Greylock Partners also
participated in the financing. Warp Drive was co-founded by Greg Verdine, a
Harvard University chemical biologist and venture partner at Third Rock, along
with Harvard University genomics expert George Church, and biolochemist James
Wells of the University of California at San Francisco.

....

Warp Drive refers to its core platform as a genomic search engine. The
companys ultimate goal is to develop the technology to the point where it will
be able to comb through naturally derived substancessuch as plants and soiland
sequence the genomes of the microbes hidden in them. Ideally, says Borisy, the
platform will be able to use that information to help scientists uncover new
molecules that have the highest probability of hitting disease targets.

Nature has been one of the drug industrys richest sources of pharmaceutical
success stories. The menu of products that originated in the wild include
diabetes drug exenatide (Byetta), derived from Gila monster saliva; heart
failure treatment digoxin, which comes from the foxglove plant; and ziconotide
(Prialt), a pain treatment from the cone snail. Nature is an incredibly
medicinal chemist, Borisy says. Nature can drug targets in ways weve never
been able to figure out how to do.

Metagenome assembly is a big part of their pipeline, according to the [job
description posted by
them](http://www.warpdrivebio.com/careers/Computational%20Biologist.pdf). Do
not laugh at 'minimum requirement'. It is likely written by a manager with
little experience in bioinformatics, who threw in as many buzzwords as he
could. Maybe [Keith Robison of Omics!Omics!
blog](http://www.blogger.com/profile/04765318239070312590), who is an employee
and author of [Kevin's GATTACA blog](http://kevin-gattaca.blogspot.com/), who
makes fun of such ridiculous job requirements, should chat.

>

Minimum Requirements:

Master's Degree and 10+ years' work experience or a Ph.D. and 6 + years
relevant work

experience in the biological sciences, computer science or computational
science

Experience with de novo assembly of microbial or metagenomic genomes from
short

read or single molecule data using tools such as Ray, MIRA, Velvet, Celera,
ALLPATHS

Experience using assembly refinement and scaffolding tools such as AMOS,
SSPACE

Experience developing and maintaining tools in Perl utilizing BioPerl and
other open

source frameworks. Python/BioPython will also be considered. Experience in

programming in R a plus.

Experience with short read data manipulation, analysis & trimming tools, e.g.
samtools,

BWA, FASTX, FLASH, Quake, KHMER, SMALT

Experience developing SQL databases and executing complex queries (joins
across many

tables, recursive joins) on such. Experience with NoSQL databases a plus.

Experience working with Linux clusters, particularly the configuring and
execution of

jobs under Sun/Oracle Grid Engine.

Experience developing interactive web tools using HTML5.

General sequence analysis background with extensive use of BLAST, HMMER,

CLUSTAL, PHYLIP and similar tools.

Excellent communication skills, including the production of clear scientific
updates using

PowerPoint and good scientific visualization skills

Able to work collaboratively in multi-disciplinary environment to accomplish
program

and company goals.

How do you go from metagenome assembly to drug discovery? It is by looking for
gene clusters having certain signatures. We do not have time to explain, but
you may start with the following two papers (among many) -

[Automated genome mining for natural
products](http://www.biomedcentral.com/1471-2105/10/185)

[The Natural Product Domain Seeker NaPDoS: A Phylogeny Based Bioinformatic
Tool to Classify Secondary Metabolite Gene Diversity](http://www.plosone.org/a
rticle/info%3Adoi%2F10.1371%2Fjournal.pone.0034064)

