---
title: PacBio Posts Slides from User Group Meeting
tags: []
categories:
- blog
---
About two months back, we reported on an user group meeting arranged by
Pacific Biosciences.
<!--more-->

[End of Short-Read Era? (Part I)](http://www.homolog.us/blogs/blog/2013/09/21
/end-illumina-era/)

[End of Short-Read Era? (Part II)](http://www.homolog.us/blogs/blog/2013/09/22
/end-short-read-era-part-ii/)

Those looking for the actual talk abstracts and slides will find the following
links helpful. We thank PacBio for providing them.

**Alex Copeland, DOE - Joint Genome Institute**

![](http://genomebiology.jgi-psf.org/Content/MGM-11.Feb2012/images/alex.jpg)

[Slides](http://programs.pacificbiosciences.com/e/1652/-pdf-Copeland-
PBUGM091813-ppsx/2trk9z/457233139)

> Copeland offered an overview of the institutes microbial and fungal
reference assembly pipeline, describing their experience with a 10x increase
in read length and total throughput in three years on the PacBio platform. He
shared that the team has shifted to a PacBio-only pipeline, and that they are
finishing genomes on the platform for less than $2,000.

**Matthew Blow, DOE - Joint Genome Institute**

[Slides](http://programs.pacificbiosciences.com/e/1652/-com-pdf-Blow-
PBUGM091813-ppsx/2trkb2/457233139)

> Blow spoke about bacterial epigenomics, an important genome component that
his team looks at with every microbe sequenced. The team is studying
methyltransferases, their link to restriction enzymes, related sequence
motifs, and sites that are unmodified. A recent analysis of global patterns in
DNA modifications in bacteria revealed that of 198 analyzed genomes, 169
(>90%) had modified DNA bases.

**Bart Weimer, UC Davis - School of Veterinary Medicine**

[Slides](http://programs.pacificbiosciences.com/e/1652/om-pdf-Weimer-
PBUGM091813-ppsx/2trkb4/457233139)

> Weimer spoke about the 100K Foodborne Pathogen Genomes project and its goal
to provide a comprehensive database that will allow users to find clinically
relevant information about new strains in outbreak situations. He was
enthusiastic about the additional information provided by PacBio sequence
data, such as methylation and phage elements both useful in tracking and
identifying pathogens.

**Lance Hepler, Center for AIDS Research, UC San Diego**

[Slides](http://programs.pacificbiosciences.com/e/1652/com-pdf-Hepler-
PBUGM091813-pdf/2trkb6/457233139)

> Hepler used the PacBio RS to study intra-host diversity in HIV-1. He
compared PacBios performance to that of 454, the platform he and his team
previously used. Hepler noted that in general, there was strong agreement
between the platforms; where results differed, he said that PacBio data had
significantly better reproducibility and accuracy.

**George Weinstock, Washington University St. Louis**

[Slides](http://programs.pacificbiosciences.com/e/1652/pdf-Weinstock-
PBUGM091813-ppsx/2trkb8/457233139)

> Weinstock discussed his overall approach to human microbiome projects,
including both targeted 16S sequencing with PacBio, as well as shotgun
sequencing of the whole sample. In a pilot project, Weinstocks team created a
mock microbiome of 24 samples with a 300-fold range of concentration; PacBio
sequencing was able to accurately identify the taxa for all 22 species where
16S amplification succeeded, yielding highly accurate full-length 16S
consensus sequences.

**John Huddleston, University of Washington**

[Slides](http://programs.pacificbiosciences.com/e/1652/pdf-Huddleston-
PBUGM091813-pdf/2trkbb/457233139)

> Huddleston is looking at challenging regions in the human genome, noting

that assembly accuracy needs to be quite high to resolve breakpoints and

reconstruct duplication architectures. His team is working with BACs to

validate the use of the PacBio platform as a faster, more cost-effective

alternative to Sanger. In one study, his team found that PacBio results

had 99.994% identity with Sanger results and showed uniform coverage

across the clone.

**Lisbeth Guethlein, Stanford University School of Medicine**

[Slides](http://programs.pacificbiosciences.com/e/1652/-pdf-Guethlein-
PBUGM091813-pdf/2trkbd/457233139)

> Guethlein looked at highly repetitive and variable regions of the

orangutan genome. Guethlein reported that PacBio managed to accomplish

in a week what I have been working on for a couple years, (with

Sanger) and the results were concordant.

**Vince Magrini, Washington University St. Louis**

[Slides](http://programs.pacificbiosciences.com/e/1652/om-pdf-Magrini-
PBUGM091813-pdf/2trkbg/457233139)

> Magrini described a proof-of-principle RNA-seq study using SMRT

Sequencing in a nematode to help elucidate transcriptional regulation and

its effect on life cycle. Using PacBio data added more than 1,500 genes

to what had been found in the reference sequence.

**Alisha K. Holloway, Gladstone Institute**

[Slides](http://programs.pacificbiosciences.com/e/1652/asandHolloway-
PBUGM091813-ppsx/2trkbj/457233139)

> Holloway presented data from transcript identification work in chicken.

Because she uses chicken to model human heart development, she needs good

annotations of RNA produced at various developmental stages to figure out

where problems arise. Unlike short-read technologies, PacBio provided

reads long enough to span entire transcripts and dramatically improved

gene annotation.

**Bobby Sebra, Mt. Sinai School of Medicine**

[Slides](http://programs.pacificbiosciences.com/e/1652/-com-pdf-Sebra-
PBUGM091813-pdf/2trkbl/457233139)

> Sebra showed how to use BluePippin size selection from Sage Science to
increase subread lengths of PacBio data. He noted that the BluePippin sizing
step also cleans up DNA quality, compensating for any drop in yield. With size
selection, Sebra said that his team could generate microbial assemblies from
data produced by a single SMRT Cell.

**Kin Fai Au, Stanford University**

(slides not currently available)

>

Au spoke about gene isoform identification and prediction in embryonic stem
cells, commenting that long reads are essential to examining these long
regions and resolving alternative splice isoforms.

**Chongyuan Luo, Salk Institute for Biological Studies**

(slides not currently available)

>

Luo from the Ecker lab spoke about studying the genome and epigenome of
several Arabidopsis thaliana strains using SMRT Sequencing. PacBio sequence
data detected 40 percent more SNPs than short-read technology, indicating that
some regions may not have been covered well enough with short reads to find
all SNPs.

**Kevin Corcoran, Pacific Biosciences**

[Slides](http://programs.pacificbiosciences.com/e/1652/-pdf-Corcoran-
PBUGM091813-ppsx/2trkbn/457233139)

> Corcoran updated attendees about the latest on our sequencing platform,
including coming advances such as polymerase photodamage protection, the new
P5-C3 chemistry offering 8,500 base average reads, three-hour movies, Quiver
for diploid sequencing, and more.

**Jonas Korlach, Pacific Biosciences**

![](http://www.pacificbiosciences.com/aboutus/executives/img/bio-jk.jpg)

[Slides](http://programs.pacificbiosciences.com/e/1652/m-pdf-Korlach--
PBUGM091813-pdf/2trkbq/457233139)

> Korlach spoke about where the PacBio platform is heading, including use for
customer projects that include large numbers of samples, higher complexity
metagenomic studies, and assemblies of larger genomes. He also mentioned
upcoming technology improvements, such as library prep automation and new data
analysis algorithms.

