---
title: Gut Metagenome in Children during Cholera and Recovery
tags: []
categories:
- blog
---
[An interesting
paper](http://www.gutpathogens.com/content/pdf/1757-4749-5-1.pdf) came out in
a journal 'Gut Pathogens'.
<!--more-->

We never heard of 'Gut Pathogens' journal. This is the first time. Also, we
have not seen any genomics paper from any Bangladeshi organization. Speaking
of scientific content, the authors only sequence amplified rRNA of gut
pathogens and compare the sequences with a reference database. We wonder what
a full metagenomic profile of gut (reassembly of all microbes) would reveal.
Here is the full abstract -

>

**Method**

Bacterial community dynamics in children suffering from cholera and during
recovery period

were examined in the present study by employing metagenomic tool, followed by
DNA

sequencing and analysis. For this, bacterial community DNA was extracted from
fecal

samples of nine clinically confirmed cholera children (age 23 years) at day 0
(acute

cholera), day 2 (antibiotic therapy), day 7 and, and day 28, and the variable
region of 16S

rRNA genes were amplified by universal primer PCR.

**Results**

454 parallel sequencing of the amplified DNA followed by similarity search of
the sequenced

data against an rRNA database allowed us to identify V. cholerae, the cause of
cholera, in all

nine children at day 0, and as predominant species in six children, accounting
for 35% of the

total gut microbiota on an average in all the nine children. The relative
abundance (mean

sem %) of bacteria belonging to phyla Proteobacteria, Firmicutes,
Bacteroidetes, and

Actinobacteria, was 55 7, 18 4, 13 4, and 8 4, respectively, at day 0, while
these

values were 12 4, 43 4, 33 3, and 12 2, respectively, at day 28. As antibiotic
therapy

began, V. cholerae count declined significantly (p< 0.001) and was found only
in four

children at day 2 and two children at day 7 with the relative abundance of
3.7% and 0.01%,

respectively, which continued up to day 28 in the two children. Compared to
acute cholera

condition (day 0), the relative abundance of Escherichia coli, Enterococcus,
and Veillonella

increased at day 2 (antibiotic therapy) while Bifidobacterium, Bacteroides,
and

Ruminococcus decreased.

**Conclusion**

Cholera results expulsion of major commensal bacteria of phyla Bacteroidetes,
Firmicutes,

and Actinobacteria, and increase of harmful Proteobacteria to colonize the gut
during acute

and convalescence states. The observed microbiota disruption might explain the
prevalent

malnutrition in children of Bangladesh where diarrheal diseases are endemic.

