---
title: Metagenomics - the Ticket is Expensive, but is the Train Going Anywhere?
tags: []
categories:
- blog
---
**Hype-drive Science**
<!--more-->

Year 2013 is turning out to be the year of frustrated realists, who cannot
tolerate the <del>hypothesis-free and data-driven</del> hype-driven science
any more. Adrian Bird, who earlier presented epigenetics as a genuine
scientific field (see [here](http://genesdev.cshlp.org/content/16/1/6.long)
and [here](http://neuron.illinois.edu/sites/default/files/U3_L7_Supplement_Per
ceptionsOfEpigenetics.pdf)), expressed his frustration with the hype [in a
recent Cell commentary](http://www.homolog.us/blogs/blog/2013/08/30/spades-
assembler-us-effect/), and added ENCODE, GWAS and long non-coding RNAs for
good measure. Now another critical piece appeared on the burgeoning field of
metagenomics from a researchers, who was more positive in the past.

![](http://depts.washington.edu/microobs/images/people/mila2.jpg)

Ludmila Chistoserdova, a researcher from University of Washington, wrote two
hopeful commentaries on metagenomics in 2010.

[Recent progress and new challenges in metagenomics for
biotechnology](http://www.ncbi.nlm.nih.gov/pubmed/20495950)

[Functional metagenomics: recent advances and future
challenges](http://www.ncbi.nlm.nih.gov/pubmed/21415887)

However, her recent commentary titled - ["Is metagenomics resolving
identification of functions in microbial communities?"](http://onlinelibrary.w
iley.com/doi/10.1111/1751-7915.12077/full) is far more critical. In a section
titled - "Did new-generation sequencing technologies really transform
metagenomics?", she wrote:

> For example, it has been concluded from the analysis of massive data
representing various mammalian metagenomes that the communities were
functionally redundant (Lozupone et?al., 2012). However, this conclusion is
most likely due to the coarse-grained nature of defining a function.

Hmmm....the authors redefined the word 'function' to make inflated claims.
Where did we [hear that before?](http://gbe.oxfordjournals.org/content/early/2
013/02/20/gbe.evt028.short)

Also her revered tone in previous reviews is replaced with subdued (and
sometimes overt) mockery, as you can see below (emphasis ours).

>

However, these days, most scientists do not have a chance to have a look at
the raw data, for the sheer volume of them, and thus they rely on the software
packages, some of which are designed to process the data all the way from raw
input to a variety of statistical analyses, expressed as either simple graphs
or very sophisticated displays (Caporaso et?al., 2010; Gibbons et?al., 2013)
**that would make modern art museums proud were such displays made in acrylic
and on large canvases**. As a biologist, I still believe that the goal of
bioinformatics is to help decipher biological meanings and trends as opposed
to be an activity on its own. Sometimes I simply gaze at some of these
displays having no idea of what they might mean. I recently reviewed a paper
that was based on such push-of-a-button analysis of microbial communities
representing dramatically different soils, including pristine versus
agricultural (nitrogen-impacted). This manuscript had the most beautiful
graphs, but they made absolutely no sense as the analyses noted no difference
between the two types of soils, whichever comparative dimension was applied.
However, a large body of prior knowledge on the effect of nitrogen onto
microbial communities exists (Ollivier et?al., 2011) that disagrees with these
automated analyses (and beautiful but useless graphs).

What happened? It seems she finally figured out the BS that is going on, just
like Adrian Bird. Her previous review of the field ("Functional Metagenomics:
Recent Advances and Future Challenges") used the word 'insight' 12 times,
which was only 1/3rd of ours here:

[Shocking Finding that a Genome by Itself Provides Little
Insight](http://www.homolog.us/blogs/blog/2013/06/05/shocking-finding-that-
genomes-by-themselves-provide-little-insight/)

In the bigger context, the researchers in metagenomics are experiencing the
same three-pronged problems that we did with our [bootstrapped genome
project](http://www.homolog.us/blogs/blog/2013/03/04/sequester-special-a
-bootstrapped-genome-project-i/), or experienced with our chlamy
transcriptomics paper discussed here ([Good Riddance Death of the Genome
Paper](http://www.homolog.us/blogs/blog/2013/05/06/good-riddance-death-of-the-
genome-paper/)).

1\. High-profile commercial journals want genome papers, because 'Genome of
Honey Bee Decoded' is far more newsworthy than a truly informative paper
analyzing the function of a gene active in the brain of queen bee or comparing
the genomes of bees, ants and other social insects.

2\. PIs want high-profile papers to pay for expensive genome projects and
sequencing.

3\. Genomes, by themselves, provide very little insight into the functions of
genes.

[In our experience](http://www.homolog.us/blogs/blog/2013/05/06/good-riddance-
death-of-the-genome-paper/), even though our chlamy paper was three years
early and provided more functional information than the genome paper, top
journals did not take it in 2004. Yet the genome paper (which incidentally did
not cite our paper) was accepted by Science in 2007 for discovering not much
from the genome.

\---------------------------------------------------------

**Warp Drive Bio**

![](http://www.warpdrivebio.com/images/pic-sub-warpdrivebio-logo.png)

Are all hopes lost on metagenomics? Maybe not. On the positive side, a company
named [Warp Drive Bio](http://www.warpdrivebio.com/about.php) that uses
metagenomics to discover drugs from natural products is back into news. The
company was founded through an innovative financial deal in the form of a call
option arrangement with the big Pharma Sanofi. Now a Harvard professor, who
founded the company, is taking three years' leave from the university to run
it as the CEO. The company is making claims that they have "two molecules
within striking distance of human studies".

[Natures Second Act](http://cen.acs.org/articles/91/i35/Natures-Second-
Act.html)

> Those watching the field might wonder if the research communitys enthusiasm
is grounded in reality. Whenever Verdine waxes too rhapsodically about the
science his company is pursuing, hes brought back to Earth by one of Warp
Drives board members, Greylock venture capitalist Bill Helman. According to
Verdine, he likes to ask, Yeah, Greg, but where are the drugs?

That question is one that scientists at the biotech firm pose to one another
on a regular basis as a playful reminder of their mission. In its first six
months, Warp Drive tested the power of its technology by rediscovering through
gene mining every known compound, plus a half-dozen new ones, in the drug
class it is pursuing, Verdine says. By the end of 2014, Warp Drive expects to
have two molecules within striking distance of human studiesthat is, with an
established mechanism of action and reasonable pharmacological properties.

Are those claims inflated as well? We do not know and some readers of 'In the
Pipeline' blog are skeptical.

[More From Warp Drive Bio (And Less From Aileron?)](http://pipeline.corante.co
m/archives/2013/07/11/more_from_warp_drive_bio_and_less_from_aileron.php)

> Even if I ran a 10billion$ R&D; unit somewhere, I would be very skeptical of
paying a significant amount of money to access their new library.

> It's funny how you can't get grant funding to go on a fishing expedition but
you can get big VC/pharma money.

Still we choose to give Warp Drive Bio the benefit of doubt, as long as Sanofi
and VCs are buying their expensive train ticket.

