---
title: Structure-based whole genome realignment reveals many novel non-coding RNAs
categories:
- ncRNA
---
[Link](http://genome.cshlp.org/content/early/2013/01/07/gr.137091.111?top=1)
<!--more-->

> Recent genome-wide computational screens that search for conservation of RNA
secondary structure in whole genome alignments (WGAs) have predicted thousands
of structural noncoding RNAs (ncRNAs). The sensitivity of such approaches,
however, is limited due to their reliance on sequence-based whole-genome
aligners, which regularly misalign structural ncRNAs. This suggests that many
more structural ncRNAs may remain undetected. Structure-based alignment, which
could increase the sensitivity, has been prohibitive for genome-wide screens
due to its extreme computational costs. Breaking this barrier, we present the
pipeline REAPR (RE-Alignment for Prediction of structural ncRNA), which
efficiently realigns whole genomes based on RNA sequence and structure, thus
allowing us to boost the performance of de novo ncRNA predictors, such as
RNAz. Key to the pipeline's efficiency is the development of a novel banding
technique for multiple RNA alignment. REAPR significantly outperforms the
widely-used predictors RNAz and EvoFold in genome-wide screens; in direct
comparison to the most recent RNAz screen on D. melanogaster, REAPR predicts
twice as many high-confidence ncRNA candidates. Moreover, modEncode RNA-Seq
experiments confirm a substantial number of its predictions as transcripts.
REAPR's advancement of de novo structural characterization of ncRNAs
complements the identification of transcripts from rapidly accumulating RNA-
Seq data.

