---
title: RECOMB 2014 Talks
tags: []
categories:
- blog
---
We earlier covered two of the RECOMB talks.
<!--more-->

a) The paper by Rayan Chikhi (Rayan Chikhi, Antoine Limasset, Shaun Jackman,
Jared Simpson and Paul Medvedev. On the representation of de Bruijn graphs)

[De Novo Assembly of Human Genome with Only 1.5 GB
RAM](http://www.homolog.us/blogs/blog/2014/01/21/de-novo-assembly-human-
genome-1-5-gb-ram/)

b) The paper by SPAdes group (Yana Safonova, Anton Bankevich and Pavel
Pevzner. DipSPAdes: Assembler for Highly Polymorphic Diploid Genomes)

[dipSpades Beats Haplomerger Hands Down in Diploid
Assembly](http://www.homolog.us/blogs/blog/2014/03/04/dipspades-beats-
haplomerger-hands-diploid-assembly/)

The following talks also appear interesting -

c) Siavash Mirarab, Nam-Phuong Nguyen and Tandy Warnow. PASTA: ultra-large
multiple sequence alignment

[Slides](http://www.cs.utexas.edu/users/tandy/benselfridge.pdf),
[Paper](http://www.cs.utexas.edu/~smirarab/papers/pasta.pdf)

>

? Divide-and-conquer approach to alignment

? Decomposes the taxa set into small subsets

? Aligns the subsets using a base alignment

method

? Merges the subset alignments into a full MSA

? Co-estimates a phylogenetic tree

d) Ngan Nguyen, Glenn Hickey, Daniel Zerbino, Brian Raney, Dent Earl, Joel
Armstrong, David Haussler and Benedict Paten. Building a Pangenome Reference
for a Population

e) Henry C.M. Leung, S.M. Yiu and Francis Chin. IDBA-MTP: A Hybrid
MetaTranscriptomic Assembler Based on Protein Information

[Code can be downloaded from here](www.cs.hku.hk/~alse/hkubrg/.)

f) Derek Aguiar, Eric Morrow and Sorin Istrail. Tractatus: an exact and
subquadratic algorithm for inferring identity-by-descent multi-shared
haplotype tracts

[download code here](http://www.brown.edu/Research/Istrail_Lab/tractatus.php)

Here is what the community likes ([according to Springer](http://www.springer.
com/computer/bioinformatics/book/978-3-319-05268-7)) -

![Capture](http://www.homolog.us/blogs/wp-
content/uploads/2014/03/Capture3-300x145.png)

We looked for the last paper, but found this one instead (same authors,
different title, different year). How different is their approach compared to
Sailfish by Petro?

[An Alignment-free Regression Approach to Estimating Allele-Speci c Expression
in F1 Animals](http://www.cs.unc.edu/~ping/MLproject.pdf)

\---------------------------------------------------------------------

The remaining ones are given below.

[Accepted Papers for RECOMB
2014](http://www.compbio.cmu.edu/recomb/acceptedpapers.html)

>

Jianzhu Ma, Sheng Wang and Jinbo Xu. MRFalign: Protein Homology Detection
through Alignment of Markov Random Fields

Siavash Mirarab, Nam-Phuong Nguyen and Tandy Warnow. PASTA: ultra-large
multiple sequence alignment

Wen-Yun Yang, Farhad Hormozdiari, Eleazar Eskin and Bogdan Pasaniuc. A
Spatial-Aware Haplotype Copying Model with Applications to Genotype Imputation

Emily Berger, Deniz Yorukoglu and Bonnie Berger. HapTree: A novel Bayesian
framework for single individual polyplotyping using NGS data

Ngan Nguyen, Glenn Hickey, Daniel Zerbino, Brian Raney, Dent Earl, Joel
Armstrong, David Haussler and Benedict Paten. Building a Pangenome Reference
for a Population

Zhanyong Wang, Jae-Hoon Sul, Sagi Snir, Jose A. Lozano and Eleazar Eskin.
Gene-Gene Interactions Detection Using A Two-stage Model

Arne Mller, Frank Bruggeman, Brett Olivier and Leen Stougie. Fast Flux Module
Detection using Matroid Theory

Shaun Mahony, Matthew Edwards, Esteban Mazzoni, Richard Sherwood, Akshay
Kakumanu, Carolyn Morrison, Hynek Wichterle and David Gifford. An integrated
model of multiple-condition ChIP-seq data reveals predeterminants of Cdx2
binding

Rui Wang and Scott Schmidler. Bayesian Multiple Protein Structure Alignment

Jianling Zhong, Todd Wasson and Alexander Hartemink. Learning protein-DNA
interaction landscapes by integrating experimental data through computational
models

Ewa Szczurek and Niko Beerenwinkel. Modeling mutual exclusivity of cancer
mutations

Marinka Zitnik and Blaz Zupan. Imputation of Quantitative Genetic Interactions
in Epistatic MAPs by Interaction Propagation Matrix Completion

Hamidreza Chitsaz and Mohammad Aminisharifabad. Exact Learning of RNA Energy
Parameters From Structure

Shutan Xu, Shuxue Zou and Lincong Wang. A geometric clustering algorithm and
its applications to structural data

Armin Tpfer, Tobias Marschall, Rowena A Bull, Fabio Luciani, Alexander
Schnhuth and Niko Beerenwinkel. Viral quasispecies assembly via maximal clique
enumeration

Yana Safonova, Anton Bankevich and Pavel Pevzner. DipSPAdes: Assembler for
Highly Polymorphic Diploid Genomes

Jan Hoinka, Alexey Berezhnoy, Zuben E. Sauna, Eli Gilboa and Teresa Przytycka.
AptaCluster - A Method to Cluster HT-SELEX Aptamer Pools and Lessons from its
Application

Mingfu Shao, Yu Lin and Bernard Moret. An Exact Algorithm to Compute the DCJ
Distance for Genomes with Duplicate Genes

Kelley Harris, Sara Sheehan, John Kamm and Yun S. Song. Decoding Coalescent
Hidden Markov Models in Linear Time

Chen-Ping Fu, Vladimir Jojic and Leonard Mcmillan. An Alignment-Free
Regression Approach for Estimating Allele-Specific Expression using RNA-Seq
Data

Henry C.M. Leung, S.M. Yiu and Francis Chin. IDBA-MTP: A Hybrid
MetaTranscriptomic Assembler Based on Protein Information

Arun Konagurthu, Parthan Kasarapu, Lloyd Allison, James Collier and Arthur
Lesk. On sufficient statistics of least-squares superposition of vector sets

Yu Zheng and Louxin Zhang. Reconciliation with Non-binary Gene Trees Revisited

Keith Noto, Carla Brodley, Saeed Majidi, Diana Bianchi and Donna Slonim. CSAX:
Characterizing Systematic Anomalies in eXpression Data

Adrian Guthals, Christina Boucher and Nuno Bandeira. The generating function
approach for peptide identification in spectral networks

Raunak Shrestha, Ermin Hodzic, Jake Yeung, Kendric Wang, Thomas Sauerwald,
Phuong Dao, Shawn Anderson, Himisha Beltran, Mark A. Rubin, Colin Collins,
Gholamreza Haffari and S. Cenk Sahinalp. HIT.nDRIVE: Multi-Driver Gene
Prioritization based on Hitting Time

Murray Patterson, Tobias Marschall, Nadia Pisanti, Leo van Iersel, Leen
Stougie, Gunnar W. Klau and Alexander Schoenhuth. WhatsHap: Haplotype Assembly
for Future-Generation Sequencing Reads

Rayan Chikhi, Antoine Limasset, Shaun Jackman, Jared Simpson and Paul
Medvedev. On the representation of de Bruijn graphs

Adam Bloniarz, Ameet Talwalkar, Jonathan Terhorst, Michael Jordan, David
Patterson, Bin Yu and Yun Song. Changepoint Analysis for Efficient Variant
Calling

Y. William Yu, Deniz Yorukoglu and Bonnie Berger. Traversing the k-mer
landscape of NGS read datasets for quality score sparsification

Hetu Kamisetty, Bornika Ghosh, Christopher James Langmead and Chris Bailey-
Kellogg. Learning Sequence Determinants of Protein:protein Interaction
Specificity with Sparse Graphical Models

Ben Raphael and Fabio Vandin. Simultaneous Inference of Cancer Pathways and
Tumor Progression from Cross-Sectional Mutation Data

Shay Zakov and Vineet Bafna. Reconstructing Breakage Fusion Bridge
architectures using noisy copy numbers

Derek Aguiar, Eric Morrow and Sorin Istrail. Tractatus: an exact and
subquadratic algorithm for inferring identity-by-descent multi-shared
haplotype tracts

Hua Wang, Heng Huang and Chris Ding. Correlated Protein Function Prediction
via Maximization of Data-Knowledge Consistency

