---
title: Reporting on the Northwest Developmental Biology Conference - (1)
tags: []
categories:
- blog
---
New readers can follow us on twitter -
[@homolog_us](https://twitter.com/homolog_us)
<!--more-->

Over the last three days, I was immersed at the Northwest Development Biology
Conference at the Friday Harbor lab. Readers may not know that the Pacific
Northwest region has very large concentration of developmental biologists
working on zebrafish. That is no surprise, because the center of zebrafish
universe is [Monte Westerfield](http://www.neuro.uoregon.edu/westerf/lab.html)
working at University of Oregon. The conference was an wonderful experience
and I was possibly the only bioinformatician allowed to enjoy the show. For
the benefit of our readers, I will cover what I enjoyed, what I learned and
what I did not learn in a series of three commentaries. The first post (this
one) presents a set of general observations. [The next
one](http://www.homolog.us/blogs/blog/2014/03/25/reporting-on-the-northwest-
developmental-biology-conference-2/) will discuss a few interesting talks and
a third brief one will mention what was missing. I will also write separate
blog posts on a handful of individual topics discussed at the conference.

\------------------------------------------------------------

**No tweet conference**

We have become used to hash-tagged conferences and often the tweets emanating
from the conferences are so informative that an online reader can reconstruct
major parts of the talks (e.g. check [Gene Myers' dazzler
assembler](http://www.homolog.us/blogs/blog/2014/02/14/dazzle-assembler-
pacbio-reads-gene-myers/)). In contrast, NWDB was twitter-free. That was not
as a policy, but because the none of the attendees was interested in tweeting
and listening at the same time. I can definitely see the wisdom of Lex
Nederbragt's post in -

[Defending no-tweeting/blogging requests at
conferences](http://flxlexblog.wordpress.com/2014/02/13/defending-no-
tweetingblogging-requests-at-conferences/)

\------------------------------------------------------------

**Biochemists apologize**

I also noticed that many biochemists apologized (for being biochemists) at the
beginning of their talks at this conference and wonder what a
bioinformatician/computational biologist would have to do !! This
'biochemistry vs biology' distinction brings up Phil Anderson's [More is
Difference](http://www.homolog.us/blogs/blog/2012/07/06/more-is-different/) in
my mind. Based on his understanding of 'broken symmetry', Anderson discussed
that every layer of science added new complexity not easily predictable from
the lower layers. For example, particle physicists cannot explain interesting
observations in condensed matter physics despite knowing all fundamental laws
of nature at the core of condensed matter physics. In the same vein,
biochemists understand all molecules in the cell, but that understanding is
not enough to explain cell biology.

Irrespective of that, the developmental biology talks from biochemists were
quite enjoyable, because they added an extra dimension in their analysis.

\------------------------------------------------------------

**Beautiful Friday Harbor lab**

Let me take a step back and describe my arrival at the conference. For those
who do not know, Friday Harbor lab (FHL) is located in an island at the
remotest corner of this country. Miss a turn or take the wrong ferry and you
will be in Canada (clicking on the image will take you to actual Google map).

![Capture](http://www.homolog.us/blogs/wp-
content/uploads/2014/03/Capture10-300x194.png)

This was my first trip to the island and I was disappointed to arrive at the
Anacortes Ferry terminal in the middle of a nasty thunderstorm. Thankfully,
the sun came out by the time I reached the island and stayed that way for the
following three days. After getting down from the Ferry, I decided to walk
from the terminal to FHL. People of the island were so nice that at least two
cars stopped to ask me where I was going and whether I would need a ride.

I took the second ride from Nicholas Gidmark, a post-doc at FHL, and he
dropped me at the main office of the lab. To my surprise, the first person I
met there was someone I knew for many years, thanks to doing some good karma
in past life :)

![](https://faculty.washington.edu/bjswalla/images/dr_billie_swalla.jpg)

Billie Swalla is a well-known marine biologist, who tracks down all kinds of
[cool animals in the ocean](https://faculty.washington.edu/bjswalla/). Her
recent genome paper (expect to see in Nature in a few months), along with
other papers on ctenophores, will indeed change the biology textbooks and not
in an ENCODE way. For readers's benefit, we wrote a [separate blog post on
comb jellies and animal
evolution](http://www.homolog.us/blogs/ncrna/2014/03/24/animal-ancestors-
sponges-or-comb-jellies/).

>

This year, the first FHL animal to have its entire genome sequenced will be in
the limelight because the genome will be published and made available soon.

.......

For years textbooks have started discussion of animals with sponges. They were
considered to lack true tissues and had a sessile life style, sitting on the
ocean floor as adults and capturing small particles. However, after sequencing
the entire genomes of several sponges, it became clear that they are more
complex than previously thought. Studies published in 2008 suggested that
ctenophores, or comb jellies, were at the base of the animal tree of life, and
it became a priority to sequence one of these species. Actually two different
species have been sequenced, which is important for comparative purposes.
These two comb jelly genomes show remarkable congruence in their genomes and
similarities to sponge genomes.

Did I mention that she is also the (interim) director of the Friday Harbor
lab?

\------------------------------------------------------------

**Cultural differences - rural versus urban, Canada vs USA**

I already mentioned about people stopping voluntarily and asking whether I
would need a ride. That is the part of trustful rural culture I experienced
previously, when I drove out of Redmond and accidentally got my car battery
down. All techies ignored my plea for help, and only two cars stopping for me
were from farmers living in rural areas.

My other observation of difference between Canada and USA is more relevant to
academia. Many Canadian researchers from British Columbia and Alberta attended
the conference and I noticed that their research labs were less reliant on
post-docs. In fact, many labs do not have a single post-docs and they are
raising competent and independent graduate students. Another ray of hope is
small US colleges like Reed college and Lewis and Clark college, where many
good researchers are joining to teach. Their undergraduate students are
getting a taste of research life at an early stage of life, when they are
['foolish enough to fail' according to
Brenner](http://kingsreview.co.uk/magazine/blog/2014/02/24/how-academia-and-
publishing-are-destroying-scientific-innovation-a-conversation-with-sydney-
brenner/).

\------------------------------------------------------------

**Everyone is angry about ENCODE**

If you read BioMickWatson's [Dear
ENCODE.](http://biomickwatson.wordpress.com/2013/02/22/dear-encode/) or Sean
Eddy's post-Dan Graur paper (discussed in ["Homolog.us Blog Calls for Sean
Eddy be Fired for the Sake of Good
Science"](http://www.homolog.us/blogs/blog/2013/04/09/homolog-us-blog-calls-
for-sean-eddy-be-fired-for-the-sake-of-good-science/)), you would think that
ENCODE was an enormously successful project that made a few 'missteps'. In
reality, it was an unnecessary waste of money. Focusing

During lunch and dinner, many participants commented on ENCODE. Those comments
were never made in a positive tone and I was surprised at the awareness. Also,
I heard that many other biologists are extremely angry about ENCODE, even
though only a handful expressed it publicly like Dan Graur. Those betting
their lives on human brain map project should start looking for alternate
careers.

\------------------------------------------------------------

**Junk science of epigenetics**

Epigenetics (cells 'remembering things' through acetylation and methylation)
is junk science or ['faddish stuff' as explained by Mark
Ptashne](http://www.fasebj.org/content/27/1/1.full.pdf+html). When cells
divide, all 'epigenetic marks' need to be redrawn and that redrawing can only
happen through expressed genes. Therefore, cells can only remember the past
through expressed genes, and that differential expression profile gets passed
on from mother cells to daughter cells. All that had been explained in the 60s
and 70s, and Ptashne's short introductory book ["A Genetic
Switch"](http://www.amazon.com/Genetic-Switch-Third-Edition-
Revisited/dp/0879697164) had been around for over 30 years. Yet, NIH is
spending plenty of money on 'epigenetics' and 'epigenomics' and that makes
some scientist look for those 'marks' everywhere.

What surprised me at the conference was that many graduate students understood
their basic cell biology so well that they clearly explained all points
against epigenetics in the same manner as Ptashne. In contrast, when I spoke
out against epigenetics in the online bioinformatics community Biostar
([Forum: Why Does Biostar Cover Questions on Epigenetics, but not Intelligent
Design?](http://www.biostars.org/p/80196/)), almost everyone objected to my
post.

\------------------------------------------------------------

**Cas9/CRISPR Revolution**

On the positive side, several researchers mentioned in private discussions,
how revolutionary the new Cas9/CRISPR system has become for genome editing.
They are finding the method so easy to use that it could be comparable to the
invention of PCR. Previous alternatives like TALEN and ZFN were quite
cumbersome. Wikipedia page on CRISPR
[agrees](http://en.wikipedia.org/wiki/CRISPR#Mouse_models) \-

> CRISPR simplifies creation of mouse models and reduces the time required to
a matter of weeks from months or longer. Knockdown of endogenous genes has
been achieved by transfection with a plasmid that contains a CRISPR area with
a spacer, which inhibits a target gene. Injecting mouse zygotes with Cas9 and
two guide RNAs was able to disable two genes with 80% efficiency. So-called
homology-directed repair involves using Cas9 to "nick" DNA, to introduce new
gene parts to the zygote.

For those of us working on non-model organisms, cas9/CRISPR is the best hope.
Therefore, we wrote a separate short introductory commentary for the benefit
of our readers -

[Saving Bacteria from Diseases and the Cas9-CRISPR Revolution
](http://www.homolog.us/blogs/chem/2014/03/24/saving-bacteria-from-diseases-
and-the-cas9-crispr-revolution/)

\------------------------------------------------------------

**Core problems for bioinformatics**

Those discussions on ENCODE, epigenetics and cas9/CRISPR brings me to the
larger field of bioinformatics, which seems to be attracting many intelligent
researchers from physics, engineering and computer science. The situation is
not too different from the late 1940s, when Schrodinger's ["What is
Life"](http://whatislife.stanford.edu/LoCo_files/What-is-Life.pdf) attracted
many physicists to life sciences. However, there is one key difference.
Schrodinger was an accomplished physicist, who wrote the book to highlight the
key scientific problems in biology and motivated others to solve them. Surely,
the commercial world benefited hugely from the discoveries, but that was a
byproduct of their scientific curiosities. Today's new entrants are misguided
by NIH to solve the wrong set of problems. For example, if you ask
bioinformaticians about what the major problems are, they will say - variant
calling, alignment, assembly, phylogeny, epigenetics, protein-interaction
network, etc. Instead I would suggest bioinformaticians to refocus and
consider the following as the key problems in bioinformatics.

1\. Understanding the DNA translation machinery.

2\. Figuring out the origin of eukaryotic cell.

3\. Figuring out the origin of multicellularity.

4\. Figuring out the vertebrate and insect body plans.

5\. Figuring out the origin of adaptive immune system.

6\. Figuring out how sex and reproduction work at the genetic level.

7\. Understanding plant body plan and evolution.

I can think of several other problems, but overall I believe it will be more
satisfying for bioinformaticians to align themselves with the core problems of
biology and then apply their specialized computational skills.

