---
title: Announcing the Results of 'Best of 2013'
tags: []
categories:
- blog
---
Dear Readers,
<!--more-->

We like to thank our judges for taking time to review the nominations for [Top
Bioinformatics Contributions of
2013](http://www.homolog.us/blogs/blog/2013/12/03/best-2013/) (summarized
[here](http://homolog.us/wiki1/index.php?title=Best_of_2013_Nomination)). The
results are summarized below. Among eight judges, Drs. Istvan Albert and
Joanna Sulkowska could not participate due to other commitments. Therefore,
the following results are from six judges - Drs. Anton Korobeynikov, Rayan
Chikhi, Heng Li, Jared Simpson, Nikolay Vyahhi and Eran Elhaik.

**Method**

We requested each judge to select 2-3 candidates [from each category of the
nominations](http://homolog.us/wiki1/index.php?title=Best_of_2013_Nomination)
that they thought would qualify as the 'Best of 2013'. Every judge acted
independently and was not biased by the opinions of other judges. They were
also asked to not restrict themselves with the nominations, but add any
important contribution they saw missing. In that way, the large list of
nominations could be shortened without the risk of missing an important
contribution, because a good contribution would be picked by one or more
judges. We indeed saw several examples of that to happen. Although BWA-MEM
paper was not nominated, one judge picked it in his short-list.

Because we had only 3 nominations in teaching tools and journals categories,
there was no point of further pruning the lists and judges were requested not
to pick shortlists from there.

\----------------------------------------------------------------

**Results**

**Research**

**i) HGAP for PacBio Assembly (4 nominations)**

[HGAP paper](http://www.nature.com/nmeth/journal/v10/n6/full/nmeth.2474.html)
received nomination from four judges.

They said -

Judge 1: "There are two aspects of a good paper: deep theory and practical
usefulness. HGAP is not particularly strong in theory, but in practice, it is
the most important work in de novo assembly."

Judge 2: "This looks like a neat idea how one can "bootstrap" out of pretty
noisy data."

Judge 3: "No breakthrough in algorithms, back to overlap-layout-consensus
assembly, but nice and stable pipeline for current technologies, which may
lead to good biological results."

Judge 4: "The high error rate of PacBio data makes it difficult to work
directly with PacBio reads. Prior to this paper most assemblies involving
PacBio data used Illumina reads to either error correct the PacBio reads or
generate an initial assembly. This paper demonstrated that complete or nearly
complete assemblies of bacterial genomes could be generated from PacBio reads
only. The impact of this paper was remarkable. It is now standard to use
PacBio for sequencing bacteria, which was not the case one year ago. In my
view this was clearly the paper of the year in the assembly field."

**ii) Theoretical Limits of Assembly Papers (2 nominations)**

This group of papers, posted [here](http://arxiv.org/abs/1301.0068) and
[here](http://arxiv.org/abs/1304.2798) at arxiv, received nominations from two
judges.

Judge 1: "The ultimate answer of "why N50 is still so low these days". Repeat
content does matter. We can show the limit practically like building de Bruijn
graph out of reference genome, but neat theory is always a neat theory :)
There are some similar papers in the list below - I missed some of them, so
cannot comment on them."

Judge 2: "I had to give my vote to this paper given that I am interested in
assembly. It is a significant theoretical contribution (there are not so many
paper of this type in 2013). This paper mixes information theory, assembly
theory, a practical algorithmic idea, all in an elegant package."

**iii) Sailfish Paper (1 nomination)**

[Link](http://arxiv.org/abs/1308.3700)

"This paper combines two nice things in my opinion. 1) It introduces a novel
technique: using minimal perfect hashing to be faster than classical mapping
for RNAseq analysis. 2) The software is very well engineered, many ideas could
benefit to other bioinformatics software (I've written a short note about it
for my colleagues, that I could make public)."

**iv) BWA-MEM (1 nomination)**

[Link](http://arxiv.org/abs/1303.3997)

"http://arxiv.org/abs/1303.3997 not in the list, BWA-MEM, it's a good and
practical algorithm and tool, sadly unpublished."

**iv) SPAdes (1 nomination)**

"SPAdes (published this year). I would like to give it a vote, but
unfortunately it is not in your short list. I think SPAdes combines
interesting theory and good practical results. It is the type of papers I
like."

**v) Ray (1 nomination)**

"http://genomebiology.com/2012/13/12/R122 Very nice tool and paper (although
it's 2012). Single-cell vs. metagenomics is still an open question for
technology, but it's fascinating to see how Ray can handle thousands of
genomes in metagenomics sequencing."

**vi) Papers from Veli Mkinen's Group (1 nomination)**

"This focuses more on the theoretical aspect. I have been enjoying works from
his group and Lam's group at Hong Kong. They directly target problems in large
scale sequence analyses."

**vii) SOAP3-dp (1 nomination)**

"SOAPdenovo2, SOAPdenovo-trans, SOAP3-dp (especially the last one)."

**viii) Plink (1 nomination)**

"Plink - the most elementary tool in population genetics and with hope for
more improvements."

**ix) Galaxy (1 nomination)**

"Galaxy - I believe in giving credit when credit is due."

**x) Questioning 'out of Africa' (1 nomination)**

"http://www.scirp.org/journal/PaperDownload.aspx?paperID=19566 - Biological
sciences should consist of few unquestionable dogmas and more theories and
models. This paper questions what slowly is becoming a dogma in population
genetics, which is the Out of Africa hypothesis."

\--------------------------------------------------------------------------

**Blogs**

**Omics Omics (Received 4 nominations)**

1\. "Keith Robison's Omics Omics."

2\. "http://omicsomics.blogspot.ru/"

3\. "There were not many posts from this blog in 2013, but I found every
single one of them to be worth reading. One of them is an open peer review of
a paper ("Assembling a Review of a Review of Assembling"). Another one gives
the reader a heads up about 2014 sequencing technology ("Peering Through the
Flowcell Glass, Darkly")."

4\. "Like many people in bioinformatics I am very interested in the
development of new sequencing technology and omicsomics covers this area very
well. As a bonus many posts are focused on assembly."

**Phil Ashton (Received 3 nominations)**

1\. "I do not read blogs and twitter that much. So, I'd not rank here. Though,
Phil's and Lior's blogs look interesting to me."

2\. "http://bitsandbugs.org/ good original posts, "TL:DR" sections are handy."

3\. "P. Ashton writes posts that I seldom see in bioinformatics: he actually
evaluates and compares data analysis techniques himself. I find this type of
content quite refreshing. See for example the posts "Validation of BWA-mem"
and "Impact of Khmer/Diginorm on SPAdes and Velvet assembly"."

**Lior Pachter (Received 2 nominations)**

1\. "Lior Patcher. He is somewhat aggressive, but overall he made reasonable
points."

2\. "I do not read blogs and twitter that much. So, I'd not rank here. Though,
Phil's and Lior's blogs look interesting to me."

**Titus Brown (Received 1 nomination)**

"http://ivory.idyll.org/blog/ (was in the best of 2012 list)"

**Phylogenomics (Received 1 nomination)**

"http://phylogenomics.blogspot.ru/ (was in the best of 2012 list)"

**Ken Weiss (Received 1 nomination)**

"Ken Weiss - exciting and insightful writing that I often find inspiring."

**Andrew Gelman (Received 1 nomination)**

http://andrewgelman.com/ - professional an interesting blog.

**Dienekes (Received 1 nomination)**

"http://dienekes.blogspot.com/ - a very professional blog that covers topic in
population genetics (mostly humans, but not necessarily) and anthropology. The
author selects studies from leading journal as well as more remote journals
and arxiv. This broad coverage in such central blog helps promote excellent
studies that otherwise would receive less attention."

**Twitter (1 nomination)**

It seems twitter itself is not very popular among the judges, because only one
judge mentioned it and picked up the following channels.

@genetics_blog

@pathogenomenick

@bioinfo

\------------------------------------------------

**Journals**

We told the judges that they did not need to shorten the set, because it had
only three entries. They were requested to add any of their favorite that is
not mentioned. A few judges decided to comment and here they are.

Judge1. "No candidates. GigaScience and PeerJ sound good but I do not have
personal experiences with them. Arxiv has been there for too long. It is not
just 2013 it starts to thrive in computational biology. BioArxiv is not as
good as Arxiv for computational biologists."

Judge2. "I would like to add http://f1000research.com/"

Judge3. "Arxiv: In 2013 I spent more time reading papers in arXiv than
anywhere else."

Original Nominations:

1\. GigaScience

2\. PeerJ

3\. arxiv and bioarxiv

\-------------------------------------------------------------------

**Teaching Tools**

We asked the judges that they did not need to shortlist from this set, because
we had only three entries. Moreover, two other judges had conflict of
interests and did not want to comment on this set. Effectively, we were down
to four judges on this set. None of the other judges commented on this
category.

Original nominations:

1\. Stepic and Rosalind

2\. Bioinformatics in Raspberry Pi

3\. Biostar

