---
title: For Today's Comic Relief
tags: []
categories:
- blog
---
Peter Kraft, 'Professor, Harvard School of Public Health - Deputy Director,
Program in Molecular and Genetic Epidemiology', wrote -
<!--more-->

![Capture](http://www.homolog.us/blogs/wp-
content/uploads/2014/04/Capture7-300x39.png)

![BkuTxu4CUAAZwp7](http://www.homolog.us/blogs/wp-
content/uploads/2014/04/BkuTxu4CUAAZwp7-300x181.jpg)

![msn_0036](http://www.homolog.us/blogs/wp-
content/uploads/2014/04/msn_0036.gif)

How is this for other reason for proposing more and larger GWAS for 'more
complex traits' - 'my salary depends on it'?

\-------------------------------------------------

What do the scientists say, when their salaries do not depend on proposing
'big science'?

1\. Bernard S. Strauss, who retired.

[Mutation and Cancer A View from
Retirement](http://www.homolog.us/blogs/blog/2014/04/05/mutation-and-cancer-a
-view-from-retirement/)

2\. Dan Graur, who "has a very low threshold for hooey, hype, hypocrisy,
postmodernism, bad statistics, ignorance of population genetics and
evolutionary biology masquerading as -omics, and hatred of any kind" -

[Creativity & Anarchy: Big Science as an Antonym of
Science](http://judgestarling.tumblr.com/post/81823224064/creativity-anarchy-
big-science-as-an-antonym-of)

3\. Ken Weiss, Evan Pugh Professor Emeritus at Penn State, [has an entire blog
dedicated to the topic](http://ecodevoevo.blogspot.com/). He wrote a book
titled Genetic Variation and Human Disease: Principles and Evolutionary
Approaches in 1993, and has been arguing against large GWAS studies since
2000, when he wrote a Nature paper titled - "How many diseases does it take to
map a gene with SNPs?" We discussed it in the following blog post, and an
extract from his paper is posted below.

[Battle over #GWAS: Ken Weiss
Edition](http://www.homolog.us/blogs/blog/2013/07/30/battle-over-gwas-ken-
weiss-edition/)

> We do not contest that genome scanning for LD can work when there is a
strong difference in the frequency of some risk genotype(s) between cases and
controls. Nikali et al. were able to map the gene responsible for a rare
recessive ataxia in a Finnish cohort using an LD-based genome scan with
microsatellite markers using only four affected individuals, demonstrating
that it can, of course, work. Similarly, the poster boy of the SNP proponents
wears a T-shirt advertising APOE ?4 and Alzheimer disease; this association is
cited as proof-of-principle for SNP-based association mapping. By using a data
set in which APOE ?4 was known to be over-represented in cases compared with
controls, it has been possible to take a set of SNPs that are in LD with the
risk allele and identify the association without studying APOE ?4 itself. This
case, stacked in favour of positive ?ndings, shows that neither coding SNPs,
nor most SNPs, nor common SNPs, nor nearby SNPs, nor single SNPs, nor a SNP in
a neighbouring gene, nor even simple haplotypes can be relied upon, a priori,
as a single, casually designed source of LD markers. What this does show is
the not-surprising fact that when there is an association with a single risk
allele, one can identify this by using multiple markers which are in LD with
this single risk allele14. How easily would genes such as BRCA1, BRCA2, PSEN1,
PSEN2 and ABCA4, which have many alleles each of which increases risk of some
common disease, be mapped by scrutinizing a set of common SNPs in a sample
from a large cosmopolitan population13,14? No one can deny that disease
pathways have been identi?ed by genetic epidemiology studies, making
contributions to our biological knowledge. So far, however, it is lifestyle
changes that have made the most impact on reduced (or increased!) incidence of
chronic disease. In those instances in which common variants affecting common
disease really do exist, it is important to ?nd them. But we regularly hear
grander promises, and it is at least fair to ask whether scaling up current
genetic approaches, which have been likened to a search for a needle in a
needle stack (a great many individually modest, effects), would be the wisest
investment when a major justi?cation is that nothing else has worked so far.

To some this is an inapt question. New genes may identify previously unknown
biochemical pathways that may lead to therapeutic pharmaceutical improvements
(even if the latter are not speci?cally genetic). Other investigators argue
that the identi?cation of even weakly predictive screening tools is suf?cient
justi?cation, or hope that a worthwhile fraction of complex diseases will have
common variants with individually modest effect but substantial attributable
risk. The level of merit in these arguments is dif?cult to evaluate, and the
track record is unclear at best; but the arguments undeniably have aspects of
rationalizing results that are not living up to expectations, sometimes based
on unrealistically optimistic models, and sometimes dominated by wishful
thinking. At the least, we should undertake the effort to assess the
underlying assertions?including of course those that we advance here?about the
degree to which common alleles with usefully important effects exist for
chronic disease; such asssessment should use focused and systematic approaches
that take the biological realities seriously.

We know these views run against momentum and heavily vested interests. Those
interests are promoted by reference to their successes (sometimes overstated,
and often argued ?rstor onlythrough the business or journalistic press rather
than peer-reviewed articles4). We expect criticism from such quarters, but
readers should be aware of the less-publicized failures and inef?ciencies.
What per cent of complex disease gene-mapping projects whose grant proposals
promised 80% power have actually been successful in identifying the disease
genes assumed to be in the data? Application of most current mapping
strategies could arguably be more effective if focused on traits that clearly
have a genetic basis, such as frank paediatric disorders (which represent a
human version of mouse knockouts, where one can assess the effects of the
complete absence of a gene product), or on candidate genes, where the genetic
or population payoff might be higher, or subsets of complex disease, in
samples from appropriately chosen populations where LD may be high and
aetiology more marked and homogeneous (or perhaps most importantly through
reductions in environmental heterogeneity). Even if such traits are
individually rare, they are numerous, and we know genetic methods work to
unravel their aetiologies.

Resistance to genetic reductionism is not new, and we know that, by expressing
these views (some might describe them as heresies), we risk being seen as
stereotypic nay-sayers. However, ours is not an argument against genetics, but
for a revised genetics that interfaces more intimately with biology.
Biological traits have evolved by noise-tolerant evolutionary mechanisms, and
a trait that doesnt manifest until long after the reproductive lifespan of
most humans throughout history is unlikely to be genetic in the traditional,
deterministic sense of the term. Most genetic studies that focus on humans are
designed, in effect, to mimic Mendels choice of experimental system, with only
two or three frequent states with strongly different effects. That certainly
enables us to characterize some of the high-penetrance tail of distribution of
the allelic effects, but as noted above these may usually be rather rare. But
in?ated claims based on this approach can divert attention from the critical
issue of how to deal with complexity on its own terms, and fuel false hopes
for simple answers to complex questions. The problems faced in treating
complex diseases as if they were Mendels peas show, without invoking the term
in its faddish sense, that complexity is a subject that needs its own
operating framework, a new twenty-?rst rather than nineteenth?or even
twentieth?century genetics.

