---
title: Tragedy of the Day - New 'Revealing' Human Transcriptome Paper at Biorxiv
tags: []
categories:
- blog
---
Someone forwarded me the link to a [new human transcriptome
paper](http://biorxiv.org/content/early/2014/10/19/010546), and as an author
[of the first genome-wide human transcriptome paper exactly 10 years
back](http://www.sciencemag.org/content/306/5705/2242), I decided to check how
far the field advanced over the decade. What I found is quite shocking and it
is abundantly clear that these ENCODE-clowns have not learned any lesson
through their public humiliation.
<!--more-->

The paper examines the transcriptomes of ~400 female twin pairs (~800
individuals), throws all kinds of statistical analysis tools and reports
absolutely nothing interesting. No new pathways, mechanisms, nothing. Based on
that summary, I thought the title would be 'comparative analysis of
transcriptomes of 800 UK twins shows nothing'. Instead the authors chose
'transcriptome sequencing reveals widespread gene-gene and gene-environment
interactions'. Oh, please !!

BS starts right from the abstract. The last statement of the abstract says -
'We uncover large GxG and GxE effects on gene expression and likely complex
phenotypes that currently remain elusive'. We pondered about the later part of
the sentence and could find absolutely no meaning whatsoever. Do the authors
mean 'we uncover complex phenotypes that currently remain elusive'? How can
you uncover something that you do not uncover anywhere in the paper? The best
explanation is that the authors wanted to include the words 'complex
phenotype', because that is what gets them money these days. Otherwise, they
could as well say 'we uncover likely the hand of God' and be as speculative.

The main text starts with -

> Gene expression is a cellular phenotype that informs about the functional
state of the cell. Gene expression by itself is a complex trait that depends
on genetic and environmental causes.

In plain English, they are saying 'gene expression' is now the new definition
of function/phenotype, and with that broadened definition of phenotype, here
comes our speculative paper. ENCODE 3 is reborn !! That is far from the
original intention of such research projects, where they were supposed to
explain real phenotypes, not redefine phenotype. However, as Sydney Brenner
explained, [the inverse problem of physiology cannot be
solved](http://www.homolog.us/blogs/blog/2014/09/02/crossroads-ii-is-it-too-
late-to-acknowledge-that-systems-biology-and-gwas-failed/) for mathematical
reasons. Hence comes the redefinition and hiding behind 'complex phenotype'
metaphors.

A high point of the paper is to go through a long list of genes, find one or
two well known members through cherry-picking process and then assign some
kind of meaningless biological story. The authors picked Epstein-Barr virus
induced 3 gene EBI2.

> One of the top hits in LCLs is the Epstein-Barr virus induced 3 gene (EBI3)
(Figure 3). That means that ASE at the EBI3 gene depends on the interaction of
cis genetic variants and an environmental factor that, likely, in this case is
the transformation process of the B cells with EBV.

By now, we have learned that all GWAS-type papers end with the obligatory
statement that the authors could not uncover something, because their sample
size was small. That later statement then becomes the first sentence in the
follow-up grant application asking for more money to do the study with 4000
twins. This paper is no exception.

> We found an example of GxE interaction on gene expression that has been
widely described in the literature (Adiponectin) supporting the validity of
our approach. However, the limitation on power due to the sample size
prevented us to discover specific associations in most other cases.

Not that lack of funding was any obstacle for their work.

> Acknowledgements: This work has been funded by the EU FP7 grant EuroBATS
(No. 259749) which also supports AAB, AB, MND, AV, TDS. AAB is also supported
by a grant from the South-Eastern Norway Health Authority, (No. 2011060). RD
is supported by the Wellcome Trust (No. 098051). The Louis-Jeantet Foundation,
Swiss National Science Foundation, Eurpean Research Council and the NIH-NIMH
GTEx grant supports ETD. TS is an NIHR senior Investigator and holder of an
ERC Advanced Principal Investigator award. JBR and HZ are supported by the
Canadian Institutes of Health Research, Fonds de Recherche Sante du Quebec,
and the Quebec Consortium for Drug Discovery. Most computations were performed
at the Vital-IT (http://www.vital-it.ch) Center for high-performance computing
of the SIB Swiss Institute of Bioinformatics. The TwinsUK study was funded by
the Wellcome Trust; EC FP7(2007-2013) and National Institute of Health
Research (NIHR). SNP Genotyping was performed by The Wellcome Trust Sanger
Institute and National Eye Institute via NIH/CIDR.

In summary, this paper is so meaningless that we suspect it will be channeled
into Nature through Magdalena Skipper.

\-----------------------------------------------------------------------------
----------------
