---
title: A Nutrient-Driven tRNA Modification Alters Translational Fidelity and Genome-wide
  Protein Coding across an Animal Genus
categories:
- ncRNA
---
[Link](http://www.plosbiology.org/article/info%3Adoi%2F10.1371%2Fjournal.pbio.
1002015)
<!--more-->

> Natural selection favors efficient expression of encoded proteins, but the
causes, mechanisms, and fitness consequences of evolved coding changes remain
an area of aggressive inquiry. We report a large-scale reversal in the
relative translational accuracy of codons across 12 fly species in the
Drosophila/Sophophora genus. Because the reversal involves pairs of codons
that are read by the same genomically encoded tRNAs, we hypothesize, and show
by direct measurement, that a tRNA anticodon modification from guanosine to
queuosine has coevolved with these genomic changes. Queuosine modification is
present in most organisms but its function remains unclear. Modification
levels vary across developmental stages in D. melanogaster, and, consistent
with a causal effect, genes maximally expressed at each stage display
selection for codons that are most accurate given stage-specific queuosine
modification levels. In a kinetic model, the known increased affinity of
queuosine-modified tRNA for ribosomes increases the accuracy of cognate codons
while reducing the accuracy of near-cognate codons. Levels of queuosine
modification in D. melanogaster reflect bioavailability of the precursor
queuine, which eukaryotes scavenge from the tRNAs of bacteria and absorb in
the gut. These results reveal a strikingly direct mechanism by which recoding
of entire genomes results from changes in utilization of a nutrient.

Author Summary

Ribosomes translate mRNA into protein using tRNAs, and these tRNAs often
translate multiple synonymous codons. Although synonymous codons specify the
same amino acid, tRNAs read codons with differing speed and accuracy, and so
some codons may be more accurately translated than their synonyms. Such
variation in the efficiency of translation between synonymous codons can
result in costs to cellular fitness. By favoring certain coding choices over
evolutionary timescales, natural selection leaves signs of pressure for
translational fidelity on evolved genomes. We have found that the way in which
proteins are encoded has changed systematically across several closely related
fruit fly species. Surprisingly, several of these changes involve two codons
both read by the same tRNA. Here we confirm experimentally that the anticodons
of these tRNAs are chemically modified (from guanine to queuosine) in vivo,
and that the levels of this modification in different species track the
differences in protein coding. Furthermore, queuosine modification levels are
known to change during fruit fly development, and we find that genes expressed
maximally during a given developmental stage have codings reflecting levels of
modification at that stage. Remarkably, queuosine modification depends upon
acquisition of its precursor, queuine, as a nutrient that eukaryotes must
obtain from bacteria through the gut. We have thus elucidated a mechanism by
which availability of a nutrient can shape the coding patterns of whole
genomes.

