---
title: Targeted Genome Editing in Drosophila and Zebrafish using CRISPR&#47;cas9
categories:
- chem
---
First the action movie :)
<!--more-->

\------------------------------------------------

A brand new paper coming out in biorxiv appears quite promising. This is not
the first demonstration of CRISPR/cas9 in Drosophila, but they spent some time
in optimizing the system.

[An optimized CRISPR/Cas toolbox for efficient germline and somatic genome
engineering in Drosophila](http://biorxiv.org/content/early/2014/03/24/003541)

![Capture](http://www.homolog.us/blogs/chem/wp-content/uploads/sites/4/2014/03
/Capture-300x147.png)

> The type II CRISPR/Cas system has recently emerged as a powerful method to
manipulate the genomes of various organisms. Here, we report a novel toolbox
for high efficiency genome engineering of Drosophila melanogaster consisting
of transgenic Cas9 lines and versatile guide RNA (gRNA) expression plasmids.
Systematic evaluation reveals Cas9 lines with ubiquitous or germline
restricted patterns of activity. We also demonstrate differential activity of
the same gRNA expressed from different U6 snRNA promoters, with the previously
untested U6:3 promoter giving the most potent effect. Choosing an appropriate
combination of Cas9 and gRNA allows targeting of essential and non-essential
genes with transmission rates ranging from 25% - 100%. We also provide
evidence that our optimized CRISPR/Cas tools can be used for offset nicking-
based mutagenesis and, in combination with oligonucleotide donors, to
precisely edit the genome by homologous recombination with efficiencies that
do not require the use of visible markers. Lastly, we demonstrate a novel
application of CRISPR/Cas-mediated technology in revealing loss-of-function
phenotypes in somatic cells following efficient biallelic targeting by Cas9
expressed in a ubiquitous or tissue-restricted manner. In summary, our
CRISPR/Cas tools will facilitate the rapid evaluation of mutant phenotypes of
specific genes and the precise modification of the genome with single
nucleotide precision. Our results also pave the way for high throughput
genetic screening with CRISPR/Cas.

Since our primary interest is in fish, we also keep the following paper in
mind.

[Efficient genome editing in zebrafish using a CRISPR-Cas
system](http://europepmc.org/articles/PMC3686313)

> Clustered regularly interspaced short palindromic repeats (CRISPR)/CRISPR-
associated (Cas) systems have evolved in bacteria and archaea as a defense
mechanism to silence foreign nucleic acids of viruses and plasmids. Recent
work has shown that bacterial type II CRISPR systems can be adapted to create
guide RNAs (gRNAs) capable of directing site-specific DNA cleavage by the Cas9
nuclease in vitro. Here we show that this system can function in vivo to
induce targeted genetic modifications in zebrafish embryos with efficiencies
comparable to those obtained using ZFNs and TALENs for the same genes. RNA-
guided nucleases robustly enabled genome editing at 9 of 11 different sites
tested, including two for which TALENs previously failed to induce
alterations. These results demonstrate that programmable CRISPR/Cas systems
provide a simple, rapid, and highly scalable method for altering genes in
vivo, opening the door to using RNA-guided nucleases for genome editing in a
wide range of organisms.

Bacteria and archaea have evolved an elegant adaptive defense mechanism which
uses clustered regularly interspaced short palindromic repeats (CRISPR),
together with CRISPR-associated (Cas) proteins, to provide acquired resistance
to invading viruses and plasmids13. The type II CRISPR/Cas system relies on
uptake of foreign DNA fragments into CRISPR loci4 and subsequent transcription
and processing of these CRISPR repeat-spacer arrays into short CRISPR RNAs
(crRNAs)5, which in turn anneal to a trans-activating crRNA (tracrRNA) and
direct sequence-specific silencing of foreign nucleic acid by Cas proteins57
(Figure 1A). Recent in vitro studies have shown that a single synthetic guide
RNA (gRNA), consisting of a fusion of crRNA and tracrRNA, can direct
Cas9-mediated cleavage of target DNA6 (Figure 1B). However, an important
question that currently remains unanswered is whether CRISPR/Cas-based systems
can have broad utility for performing genome editing in a wide variety of
whole organisms as has been shown with other technologies such as zinc finger
nucleases (ZFNs)8 or transcription activator-like effector nucleases
(TALENs)9. This capability of ZFNs and TALENs to mediate targeted in vivo
modification of genomes has enabled both genetic studies and the development
of disease models in a broad range of organisms that were previously difficult
to alter. Here we explore the abilities of customizable gRNAs and Cas9
nuclease to efficiently modify endogenous genes in vivo in zebrafish embryos
and show that this system provides a rapid and robust alternative to ZFNs and
TALENs for performing genome editing in whole organisms.

\------------------------------------------

Other important paper -

[DNA interrogation by the CRISPR RNA-guided endonuclease
Cas9](http://www.nature.com/nature/journal/v507/n7490/full/nature13011.html)

> The clustered regularly interspaced short palindromic repeats
(CRISPR)-associated enzyme Cas9 is an RNA-guided endonuclease that uses RNADNA
base-pairing to target foreign DNA in bacteria. Cas9guide RNA complexes are
also effective genome engineering agents in animals and plants. Here we use
single-molecule and bulk biochemical experiments to determine how Cas9RNA
interrogates DNA to find specific cleavage sites. We show that both binding
and cleavage of DNA by Cas9RNA require recognition of a short trinucleotide
protospacer adjacent motif (PAM). Non-target DNA binding affinity scales with
PAM density, and sequences fully complementary to the guide RNA but lacking a
nearby PAM are ignored by Cas9RNA. Competition assays provide evidence that
DNA strand separation and RNADNA heteroduplex formation initiate at the PAM
and proceed directionally towards the distal end of the target sequence.
Furthermore, PAM interactions trigger Cas9 catalytic activity. These results
reveal how Cas9 uses PAM recognition to quickly identify potential target
sites while scanning large DNA molecules, and to regulate scission of double-
stranded DNA.

\----------------------------------

Other resources -

[CRISPR-CAS9 Faq Answered](http://blog.addgene.org/crispr-cas9-faqs-answered)

[Keeping up with CRISPR-CAS9](http://blog.addgene.org/keeping-up-with-
crisprcas9)

