---
title: Personalized Cancer Screening Will Lead to Many False Diagnosis of Blood Cancer
tags: []
categories:
- blog
---
[Here](http://www.nature.com/nm/journal/vaop/ncurrent/full/nm.3733.html) is an
interesting paper that calls into question many commonly accepted notions
about GWAS and 'personalized medicine'. The authors studied somatic mutations
in blood samples of cancer patients not known to have leukemia or lymphoma,
and found many of them to carry mutations linked with blood cancer.
<!--more-->

\------------------------------------------------------------------

**Somatic mutations**

It is well-known that our genome is not static, and the cells in the body of
every individual accumulate mutations over her lifetime. Ken Weiss often
points that out in his blog post in discussing the limitations of genome-based
comparative studies for predicting common traits or disease risks. From his
recent series -

[What if Rev Jenyns had agreed? Part I. Would evolutionary theory be
different?](http://ecodevoevo.blogspot.com/2014/10/what-if-rev-jenyns-had-
agreed-part-i.html)

[What if Rev Jenyns had agreed? Part II. Would evolutionary theory be
different from a population
perspective?](http://ecodevoevo.blogspot.com/2014/10/what-if-rev-jenyns-had-
agreed-part-ii.html)

[What if Rev Jenyns had agreed? Part III. 'Group' selection in individuals,
too.](http://ecodevoevo.blogspot.com/2014/10/what-if-rev-jenyns-had-agreed-
part-iii.html)

>

**But individuals are populations too**

Let's ask something very simple: What is your 'genotype'? You began life as a
single fertilized egg with two instances of human genomes, one inherited from
each parent (here, well ignore the slight complication of mitochondrial DNA).
Two sets of chromosomes. But that was you then, not as you are now. Now, youre
a mix of countless billions of cells. Theyre countless in several ways. First,
cells in most of your tissues divide and produce two daughter cells, in
processes that continue from fertilization to death. Second, cells die. Third,
mutations occur so that each cell division introduces numerous new DNA changes
in the daughter cells. These somatic (body cell) mutations dont pass to the
next generation (unless they occur in the germline) but they do affect the
cells in which they are found.

But how do we determine your genotype? This is usually done from thousands or
millions of cellssay, by sequencing DNA extracted from a blood sample or cheek
swab. So what is usually sequenced is an aggregate of millions of instances of
each genome segment, among which there is variation. The resulting analysis
picks up, essentially, the most common nucleotides at each position. This is
what is then called your genotype and the assumption is that it represents
your nature, that is, all your cells that in aggregate make you what you are.

In fact, however, you are not just a member of a population of different
competing individuals each with their inherited genotypes. In every meaningful
sense of the word each person, too, is a i of genomes. A person's cells live
and/or compete with each other in a Darwinian sense, and his/her body and
organs and physiology are the net result of this internal variation, in the
same sense that there is an average stature or blood pressure among
individuals in a population.

[snip]

**Is somatic variation important?**

An individual is a group, or population of differing cells. In terms of the
contribution of genetic variation among those cells, our knowledge is
incomplete to say the least. From a given variant's point of view (and here we
ignore the very challenging aspect of environmental effects), there may be
some average risk--that is, phenotype among all sampled individuals with that
variant in their sequenced genome. But somatically acquired variation will
affect that variant's effects, and generally we don't yet know how to take
that into account, so it represents a source of statistical noise, or
variance, around our predictions. If the variant's risk is 5% does that mean
that 5% of carriers are at 100% risk and the rest zero? Or all are at 5% risk?
How can we tell? Currently we have little way to tell and I think manifestly
even less interest in this problem.

Cancer is a good, long-studied example of the potentially devastating nature
of somatic variation, because there is what I've called 'phenotype
amplification': a cell that has inherited (from the person's parents or the
cell's somatic ancestors) a carcinogenic genotype will not in itself be
harmful, but it will divide unconstrained so that it becomes noticeable at the
level of the organism. Most somatic mutations don't lead to uncontrolled cell
proliferation, but they can be important in more subtle ways that are very
hard to assess at present. But we do know something about them.

Evolution is a process of accumulation of variation over time. Sequences
acquire new variants by mutations in a way that generates a hierarchical
relationship, a tree of sequence variation that reflects the time order of
when each variant first arrived. Older variants that are still around are
typically more common than newer ones. This is how the individual genomes
inherited by members of a population and is part of the reason that a group
perspective can be an important but neglected aspect of our desire to relate
genotypes to traits, as discussed yesterday. Older variants are more common
and easier to find, but are unlikely to be too harmful, or they would not
still be here. Rarer variants are very numerous in our huge, recently expanded
human population. They can have strong effects but their rarity makes them
hard to analyze by our current statistical methods.

However, the same sort of hierarchy occurs during life as somatic mutations
arise in different cells at different times in individual people. Mutations
arising early in embryonic development are going to be represented in more
descendant cells, perhaps even all the cells in some descendant organ system,
than recent variants. But because recent variants arise when there are many
cells in each organ, the organ may contain a large number of very rare, but
collectively important, variants.

The mix of variants, their relative frequencies, and their distribution of
resulting effects are thus a population rather than individual phenomenon,
both in populations and individuals. Reductionist approaches done well are not
wrong, and tell us what can be told by treating individuals as single
genotypes, and enumerating them to find associations. But the reductionist
approach is only one way to consider the causal nature of life.

Our society likes to enumerate things and characterize their individual
effects. Group selection is controversial in the sense of explaining altruism,
and some versions of group selection as an evolutionary theory have well-
demonstrated failings. But properly considered, groups are real entities that
are important in evolution, and that helps account for the complexity we
encounter when we force hyper-reductionistic, individual thinking to the
exclusion of group perspectives. The same is true of the group nature of
individuals' genotypes.

The authors of Nature Medicine paper had been looking into somatic mutations
in cancer patients for several years. Here is one of their earlier papers on
the topic.

[Somatic mutations affect key pathways in lung
adenocarcinoma](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2694412/)

> Determining the genetic basis of cancer requires comprehensive analyses of
large collections of histopathologically well-classified primary tumours. Here
we report the results of a collaborative study to discover somatic mutations
in 188 human lung adenocarcinomas. DNA sequencing of 623 genes with known or
potential relationships to cancer revealed more than 1,000 somatic mutations
across the samples. Our analysis identified 26 genes that are mutated at
significantly high frequencies and thus are probably involved in
carcinogenesis. The frequently mutated genes include tyrosine kinases, among
them the EGFR homologue ERBB4; multiple ephrin receptor genes, notably EPHA3;
vascular endothelial growth factor receptor KDR; and NTRK genes. These data
provide evidence of somatic mutations in primary lung adenocarcinoma for
several tumour suppressor genes involved in other cancersincluding NF1, APC,
RB1 and ATMand for sequence changes in PTPRD as well as the frequently deleted
gene LRP1B. The observed mutational profiles correlate with clinical features,
smoking status and DNA repair defects. These results are reinforced by data
integration including single nucleotide polymorphism array and gene expression
array. Our findings shed further light on several important signalling
pathways involved in lung adenocarcinoma, and suggest new molecular targets
for treatment.

\-----------------------------------------------------------------------------
----
