---
title: Sea Anemone is Genetically Half Animal, Half Plant
categories:
- ncRNA
---
![](http://upload.wikimedia.org/wikipedia/commons/thumb/6/64/Striped_colonial_
anemone.jpg/220px-Striped_colonial_anemone.jpg)
<!--more-->

This press release ([Sea anemone is genetically half animal, half
plant](http://www.sciencedaily.com/releases/2014/03/140318113816.htm)) got us
curious about two interesting papers recently published in Genome Research.
The original genome paper of N. vectensis [sequenced by JGI](http://genome
.jgi-psf.org/Nemve1/Nemve1.home.html) came out in Science in 2007 (["Sea
Anemone Genome Reveals Ancestral Eumetazoan Gene Repertoire and Genomic
Organization"](http://www.sciencemag.org/content/317/5834/86)), but the
following new paper found something that nobody noticed before. It seems sea
anemone genome has HYL-1, a gene needed for miRNA biogenesis in plants !!

> Yehu Moran, David Fredman and Daniela Praher from the Technau team were able
to show that the microRNAs of the sea anemone depict all the hallmarks of
plant microRNAs: They have an almost perfect complementarity to their target
RNAs, which are subsequently cleaved and not inhibited like in other animals.
Moran also discovered a gene in the sea anemone, HYL-1, which is essential for
the microRNA biogenesis in plants and was never detected in any other animal
model organism before. Moreover, when one compares the sequences of microRNAs,
one microRNA with similarity to a plant microRNA as well as one microRNA with
similarity to an animal microRNA can be found. Altogether, these findings
suggest the first evolutionary link between microRNAs of plants and animals.

More detail below -

[Cnidarian microRNAs frequently regulate targets by cleavage](http://genome.cs
hlp.org/content/early/2014/03/07/gr.162503.113.abstract)

> In bilaterians, which comprise most of extant animals, microRNAs (miRNAs)
regulate the majority of messenger RNAs (mRNAs) via base-pairing of a short
sequence (the miRNA seed) to the target, subsequently promoting translational
inhibition and transcript instability. In plants, many miRNAs guide
endonucleolytic cleavage of highly complementary targets. Because little is
known about miRNA function in nonbilaterian animals, we investigated the
repertoire and biological activity of miRNAs in the sea anemone Nematostella
vectensis, a representative of Cnidaria, the sister phylum of Bilateria. Our
work uncovers scores of novel miRNAs in Nematostella, increasing the total
miRNA gene count to 87. Yet only a handful are conserved in corals and hydras,
suggesting that microRNA gene turnover in Cnidaria greatly exceeds that of
other metazoan groups. We further show that Nematostella miRNAs frequently
direct the cleavage of their mRNA targets via nearly perfect complementarity.
This mode of action resembles that of small interfering RNAs (siRNAs) and
plant miRNAs. It appears to be common in Cnidaria, as several of the miRNA
target sites are conserved among distantly related anemone species, and we
also detected miRNA-directed cleavage in Hydra. Unlike in bilaterians,
Nematostella miRNAs are commonly coexpressed with their target transcripts. In
light of these findings, we propose that post-transcriptional regulation by
miRNAs functions differently in Cnidaria and Bilateria. The similar, siRNA-
like mode of action of miRNAs in Cnidaria and plants suggests that this may be
an ancestral state.

[Evolutionary conservation of the eumetazoan gene regulatory landscape](http:/
/genome.cshlp.org/content/early/2014/03/04/gr.162529.113.abstract)

> Despite considerable differences in morphology and complexity of body plans
among animals, a great part of the gene set is shared among Bilateria and
their basally branching sister group, the Cnidaria. This suggests that the
common ancestor of eumetazoans already had a highly complex gene repertoire.
At present it is therefore unclear how morphological diversification is
encoded in the genome. Here we address the possibility that differences in
gene regulation could contribute to the large morphological divergence between
cnidarians and bilaterians. To this end, we generated the first genome-wide
map of gene regulatory elements in a nonbilaterian animal, the sea anemone
Nematostella vectensis. Using chromatin immunoprecipitation followed by deep
sequencing of five chromatin modifications and a transcriptional cofactor, we
identified over 5000 enhancers in the Nematostella genome and could validate
75% of the tested enhancers in vivo. We found that in Nematostella, but not in
yeast, enhancers are characterized by the same combination of histone
modifications as in bilaterians, and these enhancers preferentially target
developmental regulatory genes. Surprisingly, the distribution and abundance
of gene regulatory elements relative to these genes are shared between
Nematostella and bilaterian model organisms. Our results suggest that complex
gene regulation originated at least 600 million yr ago, predating the common
ancestor of eumetazoans.

