---
title: Error Correction and Assembly Complexity of Single Molecule Sequencing Reads
tags: []
categories:
- blog
---
Recently posted biorxiv paper from M. Schatz and colleagues.
<!--more-->

[Error correction and assembly complexity of single molecule sequencing reads]
(http://www.biorxiv.org/content/early/2014/06/18/006395?utm_content=buffer069a
d&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer)

> Third generation single molecule sequencing technology is poised to
revolutionize genomics by enabling the sequencing of long, individual
molecules of DNA and RNA. These technologies now routinely produce reads
exceeding 5,000 basepairs, and can achieve reads as long as 50,000 basepairs.
Here we evaluate the limits of single molecule sequencing by assessing the
impact of long read sequencing in the assembly of the human genome and 25
other important genomes across the tree of life. From this, we develop a new
data-driven model using support vector regression that can accurately predict
assembly performance. We also present a novel hybrid error correction
algorithm for long PacBio sequencing reads that uses pre-assembled Illumina
sequences for the error correction. We apply it several prokaryotic and
eukaryotic genomes, and show it can achieve near-perfect assemblies of small
genomes (< 100Mbp) and substantially improved assemblies of larger ones. All
source code and the assembly model are available open-source.

