---
title: " At #BoG14, Posters Are Far More Interesting than Talks"
tags: []
categories:
- blog
---
You can find the abstracts
[here](http://meetings.cshl.edu/abstracts/genome2014_absstat.html). There are
~50 talks and 324 posters. In twitter, follow #BOG14 hashtag to learn what is
going on.
<!--more-->

\-------------------------------------------------------

This is a conference run by the ENCODE gang, which means almost all talks are
from their F&F;, whereas most good abstracts are hidden in the posters.

A few interesting titles -

**Meltz Steinberg, K.N. - A single haplotype platinum genome assembly **

Since Bog does not give out abstract, we will borrow the [ASHG abstract from
the same
group](http://www.ashg.org/2013meeting/abstracts/fulltext/f130122572.htm)
(Eichler at UW).

> The human genome reference sequence has provided a foundation for studies of
genome structure, human variation, evolutionary biology and human disease.
Many of these studies have also revealed, however, that there are regions of
the human reference genome that are not represented optimally. At the time the
reference genome was completed it was clear that there were some loci
recalcitrant to closure with the technology and resources available at that
time. It was not clear, however, the degree to which structural variation and
diversity affected our ability to produce a truly representative genome
sequence at these loci. Many of these regions in the genome, particularly the
structural variant loci, are often associated with repetitive sequences. In
order to discriminate between repeat copy and allelic copies, the sequence
from a single haplotype across these regions is necessary. To this end, we
have utilized a hydatidiform mole source, CHM1 to finish highly complex,
repetitive regions to high quality. Our aim is to develop a single allelic
representation of the entire human genome, the platinum reference. In order to
achieve this, we have generated ~100X whole genome shotgun sequence as
Illumina paired end data, as well as over 450 BAC sequences from the CHM1
libraries. The whole genome data has been assembled using a reference-guided
assembly and the finished BAC sequences have been incorporated into this
assembly. We have compared the CHM1 assembly to the current reference, GRCh37
to identify single nucleotide variants, structural variants, and missing
sequence from the reference. In addition, we have aligned the CHM1 Illumina
sequence to the CHM1 assembly to evaluate the efficacy of our assembly
strategy.

\-----------------------------------------------------

**Bachtrog, D. - Numerous transitions of sex chromosome in diptera**

**Bachtrog, D. - The genomic landscape of incipient species reveals a candidate behavioral isolation gene in Drosophila athabasca**

[Reversal of an ancient sex chromosome to an autosome in Drosophila](http://ww
w.nature.com/nature/journal/v499/n7458/full/nature12235.html?WT.ec_id=NATURE-2
0130718)

> Although transitions of sex-determination mechanisms are frequent in species
with homomorphic sex chromosomes1, 2, 3, heteromorphic sex chromosomes are
thought to represent a terminal evolutionary stage owing to chromosome-
specific adaptations such as dosage compensation or an accumulation of sex-
specific mutations1, 4. Here we show that an autosome of Drosophila, the dot
chromosome, was ancestrally a differentiated X chromosome. We analyse the
whole genome of true fruitflies (Tephritidae), flesh flies (Sarcophagidae) and
soldier flies (Stratiomyidae) to show that genes located on the dot chromosome
of Drosophila are X-linked in outgroup species, whereas Drosophila X-linked
genes are autosomal. We date this chromosomal transition to early drosophilid
evolution by sequencing the genome of other Drosophilidae. Our results reveal
several puzzling aspects of Drosophila dot chromosome biology to be possible
remnants of its former life as a sex chromosome, such as its minor feminizing
role in sex determination5 or its targeting by a chromosome-specific
regulatory mechanism6. We also show that patterns of biased gene expression of
the dot chromosome during early embryogenesis, oogenesis and spermatogenesis
resemble that of the current X chromosome. Thus, although sex chromosomes are
not necessarily evolutionary end points and can revert back to an autosomal
inheritance, the highly specialized genome architecture of this former X
chromosome suggests that severe fitness costs must be overcome for such a
turnover to occur.

\-----------------------------------------------------

**Bracht, J.R. - Chromosome fusions triggered by noncoding RNA**

[Genomes on the Edge: Programmed Genome Instability in
Ciliates](http://www.sciencedirect.com/science/article/pii/S0092867413000068)

> Ciliates are an ancient and diverse group of microbial eukaryotes that have
emerged as powerful models for RNA-mediated epigenetic inheritance. They
possess extensive sets of both tiny and long noncoding RNAs that, together
with a suite of proteins that includes transposases, orchestrate a broad
cascade of genome rearrangements during somatic nuclear development. This
Review emphasizes three important themes: the remarkable role of RNA in
shaping genome structure, recent discoveries that unify many deeply diverged
ciliate genetic systems, and a surprising evolutionary sign change in the role
of small RNAs between major species groups.

\-----------------------------------------------------

**Brawand, D. - Genomes of 5 African cichlid fishGene duplication as a substrate for adaptive radiation**

\-----------------------------------------------------

**Eckalbar, W.L. - Genomic characterization of the developing bat wing in the Natal long-fingered bat, Miniopterus natalensis**

\-----------------------------------------------------

**Engel, S.R. - Transcriptional regulation and protein complexes in budding yeast**

\-----------------------------------------------------

**Giacomello, S. - Spatial single-cell transcriptomics to study evolutionary differences between angiosperms and gymnosperms**

\-----------------------------------------------------

**Hancock, A. - Arabidopsis in the Cape Verde IslandsA model for understanding complex trait evolution during adaptive range expansion**

\-----------------------------------------------------

**Lamichhaney, S. - The evolutionary history of Darwins finches revealed by whole genome sequencing**

\-----------------------------------------------------

**Schatz, M.C. - Near perfect de novo assemblies of eukaryotic genomes using PacBio long read sequencing**

\-----------------------------------------------------

**Smith, J.J. - The sea lamprey meiotic map resolves ancient vertebrate genome duplications**

\-----------------------------------------------------

**Venkatesh, B. - Sequencing and comparative analysis of the elephant shark genomeNew insights into gnathostome evolution**

\-----------------------------------------------------

**Worley, K.C. - Better genomes using long reads and PBJelly 2**

\-----------------------------------------------------

**Yue, J. - A genomic study of Asymmetron lucayanum offers insight into the evolution of cephalochordates**

\-----------------------------------------------------------------------------
-

What could we have learned, if some of those posters were turned into talks?
We can give two examples.

1\. J. J. Smith works on lamprey, which has a very interesting adaptive immune
system. For example, check "[Lamprey immunity is far from primitive -
Rast](http://www.pnas.org/content/110/15/5746.short)". By learning about
genomic differences between lamprey and other vertebrates, we can learn about
how the human immune system evolved.

2\. Venkatesh, along with Sidney Brenner, works on elephant shark genome. Here
is [their project
description](http://www.imcb.a-star.edu.sg/php/venkatesh.php), which is far
better than ENCODE's goal of finding function in 100% of human genome.

> We are using a comparative genomics approach to better understand the
structure, function and evolution of the human genome. Our group is one of the
pioneers in the field of comparative genomics. We proposed the compact genome
of the fugu (Takifugu rubripes) as a model vertebrate genome in 1993 (Nature
366: 265-268, 1993) and determined its whole genome sequence in 2002 (Science
297: 1301-1310, 2002). Fugu genome was the first vertebrate genome to be
sequenced soon after the completion of the human genome. It is being widely
used as a reference genome for comparative analysis of human and other
vertebrate genomes. More recently, we identified elephant shark (Callorhinchus
milii) as having the smallest genome among cartilaginous fishes and initiated
the Elephant Shark Genome Project (PLoS Biol 5(4): e101, 2007). Cartilaginous
fishes are the oldest living group of jawed vertebrates (gnathostomes) and
serve as a critical reference for understanding the evolution of vertebrate
genomes. We have also initiated a genome project for a model jawless
vertebrate, the Japanese lamprey (Lethenteron japonicum), which has a smaller
genome than the sea lamprey. Jawless vertebrates (cyclostomes), comprising
lampreys and hagfishes, are the most basally branching lineage of vertebrates
and hence an important reference for understanding the origin and evolution of
vertebrates. In addition, we are exploring other model vertebrate genomes such
as coelacanth, gar and cichlid fishes that can contribute to our understanding
of human and other vertebrate genomes. Our group is also participating in
Genome 10K, an international project which aims to sequence the genomes of
10,000 vertebrates.

