---
title: Transcriptome in Vivo Analysis (TIVA) of Spatially Defined Single Cells in
  Live Tissue
categories:
- rnaseq
---
One reason for our splitting the RNAseq channel is to be able to write about
relevant experimental methods in addition to bioinformatics. Readers will find
[a recent paper (Jan) in Nature
Methods](http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2804.html)
interesting (h/t: gunguez).
<!--more-->

> Transcriptome profiling of single cells resident in their natural
microenvironment depends upon RNA capture methods that are both noninvasive
and spatially precise. We engineered a transcriptome in vivo analysis (TIVA)
tag, which upon photoactivation enables mRNA capture from single cells in live
tissue. Using the TIVA tag in combination with RNA sequencing (RNA-seq), we
analyzed transcriptome variance among single neurons in culture and in mouse
and human tissue in vivo. Our data showed that the tissue microenvironment
shapes the transcriptomic landscape of individual cells. The TIVA methodology
is, to our knowledge, the first noninvasive approach for capturing mRNA from
live single cells in their natural microenvironment.

![Capture](http://www.homolog.us/blogs/rnaseq/wp-
content/uploads/sites/10/2014/04/Capture-300x183.png)

The authors use an tag that captures mRNA only after photo-activation. That
way, there is no need to do physical surgery on the cells and one can capture
expressed genes in a non-invasive way.

> Here we describe a method for isolating mRNA from a single cell in complex
tissues using a photoactivatable mRNA capture molecule called the TIVA tag. We
demonstrate the utility of the TIVA tag in both cell culture and brain tissue
for capture of singlecell mRNA for subsequent RNA-seq transcriptome analysis.
We apply this approach to study the unique transcriptional landscape of single
neurons in vivo and how their transcriptomes differ fundamentally from those
of cells in culture.

