---
title: Sponge Jelly Genome is Studied to Find Evolutionary Origins of Neural Systems
categories:
- ncRNA
---
Two months back, we mentioned about an upcoming paper, where Friday Harbor lab
was a collaborator ("[Animal Ancestors Sponges or Comb
Jellies?](http://www.homolog.us/blogs/ncrna/2014/03/24/animal-ancestors-
sponges-or-comb-jellies/)"). It just [came out in Nature](http://www.nature.co
m/nature/journal/vaop/ncurrent/full/nature13400.html) today.
<!--more-->

> The origins of neural systems remain unresolved. In contrast to other basal
metazoans, ctenophores (comb jellies) have both complex nervous and mesoderm-
derived muscular systems. These holoplanktonic predators also have
sophisticated ciliated locomotion, behaviour and distinct development. Here we
present the draft genome of Pleurobrachia bachei, Pacific sea gooseberry,
together with ten other ctenophore transcriptomes, and show that they are
remarkably distinct from other animal genomes in their content of neurogenic,
immune and developmental genes. Our integrative analyses place Ctenophora as
the earliest lineage within Metazoa. This hypothesis is supported by
comparative analysis of multiple gene families, including the apparent absence
of HOX genes, canonical microRNA machinery, and reduced immune complement in
ctenophores. Although two distinct nervous systems are well recognized in
ctenophores, many bilaterian neuron-specific genes and genes of classical
neurotransmitter pathways either are absent or, if present, are not expressed
in neurons. Our metabolomic and physiological data are consistent with the
hypothesis that ctenophore neural systems, and possibly muscle specification,
evolved independently from those in other animals.

[Our earlier blog post](http://www.homolog.us/blogs/ncrna/2014/03/24/animal-
ancestors-sponges-or-comb-jellies/) has quite a bit of details on this
fascinating marine organism called comb jelly, but the Nature paper has a
figure to put it in evolutionary context.

![](http://www.nature.com/nature/journal/vaop/ncurrent/images/nature13400-f2.j
pg)

The goal here is to go further and further behind in evolutionary time-frame
and use genome comparison to find out the origins of various gene families.
One difficulty in that approach is the one has to be absolutely sure that a
gene declared as missing is not missing due to assembly artifacts. Therefore,
we read the genome assembly section very carefully, and have a [separate post
about their assembly methods in our bioinformatics
blog](http://www.homolog.us/blogs/blog/2014/05/22/what-approach-do-
researchers-take-to-avoid-short-read-noise/). The paper also included
extensive amount of transcriptome data in addition to the genome to back its
claim.

