---
title: Non-random DNA fragmentation in next-generation sequencing
categories:
- chem
---
[link](http://www.nature.com/srep/2014/140331/srep04532/full/srep04532.html?WT
.ec_id=SREP-631-20140401)
<!--more-->

> Next Generation Sequencing (NGS) technology is based on cutting DNA into
small fragments, and their massive parallel sequencing. The multiple
overlapping segments termed reads are assembled into a contiguous sequence. To
reduce sequencing errors, every genome region should be sequenced several
dozen times. This sequencing approach is based on the assumption that genomic
DNA breaks are random and sequence-independent. However, previously we showed
that for the sonicated restriction DNA fragments the rates of double-stranded
breaks depend on the nucleotide sequence. In this work we analyzed genomic
reads from NGS data and discovered that fragmentation methods based on the
action of the hydrodynamic forces on DNA, produce similar bias. Consideration
of this non-random DNA fragmentation may allow one to unravel what factors and
to what extent influence the non-uniform coverage of various genomic regions.

