---
title: The Most Difficult Problem in Computational Biology
tags: []
categories:
- blog
---
The real revolution in computational biology is taking place without much
public attention. Temple Smith, who is famous for Smith-Waterman algorithm, is
using computational methods to work on what Francis Crick called the most
difficult problem in genetics - the evolutionary origin of ribosome and
translational apparatus. This 'translation' problem got buried inside the
genetic code, but Woese argued that its evolutionary aspect needs to be
understood properly to get full understanding of the evolution of organisms
(including large eukaryotic mammals).
<!--more-->

[The Evolution of the Ribosome and the Genetic Code](http://www.ncbi.nlm.nih.g
ov/pmc/articles/PMC4187167/pdf/life-04-00227.pdf)

> The evolution of the genetic code is mapped out starting with the aminoacyl
tRNA-synthetases and their interaction with the operational code in the tRNA
acceptor arm. Combining this operational code with a metric based on the
biosynthesis of amino acids from the Citric acid, we come to the conclusion
that the earliest genetic code was a Guanine Cytosine (GC) code. This has
implications for the likely earliest positively charged amino acids. The
progression from this pure GC code to the extant one is traced out in the
evolution of the Large Ribosomal Subunit, LSU, and its proteins; in particular
those associated with the Peptidyl Transfer Center (PTC) and the nascent
peptide exit tunnel. This progression has implications for the earliest
encoded peptides and their evolutionary progression into full complex
proteins.

Readers are encouraged to check other recent papers of Smith written in
collaboration with Hartman (such as [GTPase and the origin of the ribosome](ht
tp://www.ncbi.nlm.nih.gov/pmc/articles/PMC2881122/pdf/1745-6150-5-36.pdf)).

Speaking of relevance of translation to the development of mammals, readers
may check work from Maria Barna at Stanford -

[RNA regulons in Hox 5? UTRs confer ribosome specificity to gene regulation](h
ttp://www.nature.com/nature/journal/v517/n7532/abs/nature14010.html)

> Emerging evidence suggests that the ribosome has a regulatory function in
directing how the genome is translated in time and space. However, how this
regulation is encoded in the messenger RNA sequence remains largely unknown.
Here we uncover unique RNA regulons embedded in homeobox (Hox) 5? untranslated
regions (UTRs) that confer ribosome-mediated control of gene expression. These
structured RNA elements, resembling viral internal ribosome entry sites
(IRESs), are found in subsets of Hox mRNAs. They facilitate ribosome
recruitment and require the ribosomal protein RPL38 for their activity.
Despite numerous layers of Hox gene regulation, these IRES elements are
essential for converting Hox transcripts into proteins to pattern the
mammalian body plan. This specialized mode of IRES-dependent translation is
enabled by an additional regulatory element that we term the translation
inhibitory element (TIE), which blocks cap-dependent translation of
transcripts. Together, these data uncover a new paradigm for ribosome-mediated
control of gene expression and organismal development.

Speaking of difficult problems, interestingly Smith also has his eye on origin
of the cilia and eukaryotes. For a full list of other problems, readers may
take a look at -

[Crossroads (iii) a New Direction for Bioinformatics with Twelve Fundamental
Problems](http://www.homolog.us/blogs/blog/2014/10/14/crossroads-iii-a-new-
direction-for-bioinformatics-with-twelve-fundamental-problems/)

