---
title: Two Excellent Reviews on Metagenomics
tags: []
categories:
- blog
---
Over the last few days, we have been going through various analysis tools for
analysis of metagenomics data sets. The readers working in the same field may
find the following two papers useful.
<!--more-->

Table 1 of this paper compiles the list of many useful programs.

[Metagenomics: Tools and Insights for Analyzing Next-Generation Sequencing
Data Derived from Biodiversity
Studies](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426941/)

> Advances in next-generation sequencing (NGS) have allowed significant
breakthroughs in microbial ecology studies. This has led to the rapid
expansion of research in the field and the establishment of metagenomics,
often defined as the analysis of DNA from microbial communities in
environmental samples without prior need for culturing. Many metagenomics
statistical/computational tools and databases have been developed in order to
allow the exploitation of the huge influx of data. In this review article, we
provide an overview of the sequencing technologies and how they are uniquely
suited to various types of metagenomic studies. We focus on the currently
available bioinformatics techniques, tools, and methodologies for performing
each individual step of a typical metagenomic dataset analysis. We also
provide future trends in the field with respect to tools and technologies
currently under development. Moreover, we discuss data management,
distribution, and integration tools that are capable of performing comparative
metagenomic analyses of multiple datasets using well-established databases, as
well as commonly used annotation standards.

\-----------------------------

The following paper is not a review, but rather benchmarking of a number of
analysis tools.

[An evaluation of the accuracy and speed of metagenome analysis
tools](http://biorxiv.org/content/early/2015/04/09/017830)

> Metagenome studies are becoming increasingly widespread, yielding important
insights into microbial communities covering diverse environments from
terrestrial and aquatic ecosystems to human skin and gut. With the advent of
high-throughput sequencing platforms, the use of large scale shotgun
sequencing approaches is now commonplace. However, a thorough independent
benchmark comparing state-of-the-art metagenome analysis tools is lacking.
Here, we present a benchmark where the most widely used tools are tested on
complex, realistic data sets. Our results clearly show that the most widely
used tools are not necessarily the most accurate, that the most accurate tool
is not necessarily the most time consuming, and that there is a high degree of
variability between available tools. These findings are important as the
conclusions of any metagenomics study are affected by errors in the predicted
community composition. Data sets and results are freely available from
http://www.ucbioinformatics.org/metabenchmark.html.

