---
title: Tardigrade Mystery Solved?
tags: []
categories:
- blog
---
[Tardigrades (water bears)](https://en.wikipedia.org/wiki/Tardigrade) are
fascinating animals. Unlike other animals, they can survive extreme stress
including high temperature or outer space.
<!--more-->

> Tardigrades are notable for being perhaps the most durable of known
organisms; they are able to survive extreme conditions that would be rapidly
fatal to nearly all other known life forms. They can withstand temperature
ranges from ?458 F (?272.222 C) to 300 F (149 C),[7] pressures about six times
greater than those found in the deepest ocean trenches, ionizing radiation at
doses hundreds of times higher than the lethal dose for a human, and the
vacuum of outer space.[8] They can go without food or water for more than 10
years, drying out to the point where they are 3% or less water, only to
rehydrate, forage, and reproduce.[3][9][10][11]

![Water Bear](http://www.homolog.us/blogs/wp-
content/uploads/2015/12/tardigrade_eyeofscience_1024-300x259.jpg)

How do they manage to do that? A [new
study](http://uncnews.unc.edu/2015/11/23/a-huge-chunk-of-a-tardigrades-genome-
comes-from-foreign-dna/) seemed to have found the answer. They discovered that
tardigrade genome went through large-scale horizontal gene transfer, and now
includes many genes from bacteria. Those bacterial genes possibly help
tardigrades survive extreme environment.

> The work, publish today in the Proceeding of the National Academy of
Sciences, not only raises the question of whether there is a connection
between foreign DNA and the ability to survive extreme environments, but
further stretches conventional views of how DNA is inherited.

First author Thomas Boothby, Goldstein and their collaborators revealed that
tardigrades acquire about 6,000 foreign genes primarily from bacteria, but
also from plants, fungi and Archaea, through a process called horizontal gene
transfer the swapping of genetic material between species as opposed to
inheriting DNA exclusively from mom and dad. Previously another microscopic
animal called the rotifer was the record-holder for having the most foreign
DNA, but it has about half as much as the tardigrade. For comparison, most
animals have less than one percent of their genome from foreign DNA.

Animals that can survive extreme stresses may be particularly prone to
acquiring foreign genes and bacterial genes might be better able to withstand
stresses than animal ones, said Boothby, a postdoctoral fellow in Goldsteins
lab. After all, bacteria have survived the Earths most extreme environments
for billions of years.

Mystery solved? Not so fast. A [newer
study](http://biorxiv.org/content/early/2015/12/01/033464) reported - "We
compare our assembly to a recently published one for the same species and do
not find support for massive horizontal gene transfer."

> Tardigrades are meiofaunal ecdysozoans and are key to understanding the
origins of Arthropoda. We present the genome of the tardigrade Hypsibius
dujardini, assembled from Illumina paired and mate-pair data. While the raw
data indicated extensive contamination with bacteria, presumably from the gut
or surface of the animals, careful cleaning generated a clean tardigrade
dataset for assembly. We also generated an expressed sequence tag dataset, a
Sanger genome survey dataset and used these and Illumina RNA-Seq data for
assembly validation and gene prediction. The genome assembly is ~130 Mb in
span, has an N50 length of over 50 kb, and an N90 length of 6 kb. We predict
23,031 protein-coding genes in the genome, which is available in a dedicated
genome browser at http://www.tardigrades.org. We compare our assembly to a
recently published one for the same species and do not find support for
massive horizontal gene transfer. Additional analyses of the genome are
ongoing.

Another case of ["Shocking Finding that a Genome by Itself Provides Little
Insight"](http://www.homolog.us/blogs/blog/2013/06/05/shocking-finding-that-
genomes-by-themselves-provide-little-insight/) ? Maybe two genomes together do
provide some insight. The second paper wrote -

>

Boothby et al 6 recently published a genome assembly for H. dujardini (here
called the UNC

assembly) based on a subculture of the same Sciento stock as our tardigrades.
Surprisingly, the genome is very different in span and gene content (Table 3).
We thus explored the differences between our nHd.2.3 assembly and the Boothby
et al. one to identify likely reasons for the discrepancies. The UNC H.
dujardini genome was reported to contain a surprisingly high proportion (17%)
of putatively horizontally transferred protein-coding genes, and so we also
compared the putative HGT gene set predictions with ours.

...

A significant proportion of the UNC scaffolds had zero or very low coverage of
reads mapped in both Edinburgh and UNC raw data. These scaffolds contain
matches to Bacteriodetes, spanning 27.7 Mb, that included most of the largest
scaffolds in the UNC assembly. Bacterial genomes in low complexity metagenomic
datasets often assemble with greater contiguity than does the target metazoan
genome, because bacterial DNA usually has higher per-base complexity (i.e.
does not contain so many repeats). A second group of bacterial contigs, that
appeared to derive from Proteobacteria, had a wide dispersion of coverage,
from ~10 fold higher than the H. dujardini nuclear mean to zero. Again these
are likely to derive from one or more genomes (they span 21.5 Mb). Most of the
Proteobacteria scaffolds all had distinct GC%, and grouped separately from the
true H. dujardini scaffolds. Comparing coverage between read sets, it is
striking that many of the putatively bacterial scaffolds had zero coverage in
both UNC and Edinburgh data. We presume that these scaffolds were assembled
from UNC data from other libraries (we have not yet screened their 500 and 800
base libraries) containing additional contaminants. Scaffolds with some
coverage in the UNC 300 data often had zero coverage in Edinburgh data. The
wide spread of proteobacterial scaffolds suggests some sharing of contaminants
between the UNC and Edinburgh cultures, but it is likely that these are
different taxa, as the coverages vary widely between datasets. It is thus
unlikely that these are common symbionts.

....

**These analyses thus positively identified much of the UNC assembly as likely derived from contaminant bacteria**... 

Oh well !

