---
title: Guangzhou Researcher Debunks 'Duon' Discovery of Stamatoyannopoulos
tags: []
categories:
- blog
---
Dr. John Stamatoyannopoulos, who receives [University of Washington Alumni
Early Achievement Award](http://www.homolog.us/blogs/blog/2013/07/29/uw-
alumni-early-achievement-award-2013-dr-john-stamatoyannopoulos/) for his
ENCODE 'discoveries' surprised the world again in late 2013 by [discovering
second genetic code](http://www.homolog.us/blogs/blog/2013/12/13/homolog-us-
blog-congratulates-encodes-stamatoyannopoulos-rediscovering-enhancers/) and
publishing in a glam journal.
<!--more-->

Not-at-all glam journal Molecular Biology and Evolution reports (emphasis
ours) -

[Reassessing the Duon Hypothesis of Protein
Evolution](http://mbe.oxfordjournals.org/content/32/4/1056.abstract)

![20121110122148199030](http://www.homolog.us/blogs/wp-
content/uploads/2015/04/20121110122148199030.jpg)

> There are two distinct types of DNA sequences, namely coding sequences and
regulatory sequences, in a genome. A recent study of the occupancy of
transcription factors (TFs) in human cells suggested that protein-coding
sequences also serve as the codes of TF occupancy, and proposed a duon
hypothesis in which up to 15% of codons of human protein genes are constrained
by the additional coding requirements that regulate gene expression. This
hypothesis challenges our basic understanding on the human genome. **We
reanalyzed the data and found that the previous study was confounded by
ascertainment bias related to base composition. Using an unbiased comparison
in which G/C and A/T sites are considered separately, we reveal a similar
level of conservation between TF-bound codons and TF-depleted codons,
suggesting largely no extra purifying selection provided by the TF occupancy
on the codons of human genes.** Given the generally short binding motifs of
TFs and the open chromatin structure during transcription, we argue that the
occupancy of TFs on protein-coding sequences is mostly passive and
evolutionarily neutral, with to-be-determined functions in the regulation of
gene expression.

Observations -

1\. 'Not at all glam' equals 'respected', when it comes to science and
journals. Molecular Biology and Evolution was founded by Walter Fitch and
[Masatoshi Nei](http://bio.psu.edu/directory/nxm2), two well-respected
evolutionary biologists, and continues to publish serious papers.

2\. The only 'press release' this debunking received is [a mention in Dan
Graur's blog](http://judgestarling.tumblr.com/post/115614563761/a-moment-of-
silence-in-memory-of-duons-the). Duon 'discovery' was covered by all major
media including [Times](http://science.time.com/2013/12/13/second-code-
uncovered-inside-the-dna/),
[Forbes](http://www.forbes.com/sites/emilywillingham/2013/12/13/dont-be-duped-
by-duon-dna-hype/) and possibly even Playboy.

3\. The debunkers work at a Chinese university in the state located right
across the border from Hong Kong. Guangzhou is the capital of Guandong, which
used to be one of the poorest provinces of China 25 years back. Then the
economic liberalization made it the fastest growing manufacturing center, and
now the place is focusing on building a knowledge economy. Shenzhen is the
more well-known city of the province.

4\. Professor Xionglei He, the senior author of the paper, did his PhD at
University of Michigan with Jiangzhi Zhang, who himself was a PhD student of
Masatoshi Nei. Masatoshi Nei is a colleague of [Ken
Weiss](http://ecodevoevo.blogspot.com/) at Penn State University and
supervised Dan Graur's PhD thesis in the 80s. So, you can clearly see the
pattern that USA/UK are sending away highly trained scientists to China, while
feeding the clowns like Birney and Stamatoyannopoulos. Rinse and repeat for a
few years and you get -

[Homolog.us Reports from Asia Will Hong Kong be the Bioinformatics Capital of
the World?](Homolog.us Reports from Asia  Will Hong Kong be the Bioinformatics
Capital of the World?)

5\. We bring back the obligatory video that is reserved for Dr.
Stamatoyannopoulos.

<iframe width="560" height="315" src="http://www.youtube.com/embed/HttF5HVYtlQ" frameborder="0"> </iframe>

