---
title: Reducing Assembly Complexity of Microbial Genomes with Single-molecule Sequencing
tags: []
categories:
- blog
---
[Link](http://www.genomebiology.com/2013/14/9/R101)
<!--more-->

> Background

The short reads output by first- and second-generation DNA sequencing
instruments cannot completely reconstruct microbial chromosomes. Therefore,
most genomes have been left unfinished due to the significant resources
required to manually close gaps in draft assemblies. Third-generation, single-
molecule sequencing addresses this problem by greatly increasing sequencing
read length, which simplifies the assembly problem.

Results

To measure the benefit of single-molecule sequencing on microbial genome
assembly, we sequenced and assembled the genomes of six bacteria and analyzed
the repeat complexity of 2,267 complete bacteria and archaea. Our results
indicate that the majority of known bacterial and archaeal genomes can be
assembled without gaps, at finished-grade quality, using a single PacBio RS
sequencing library. These single-library assemblies are also more accurate
than typical short-read assemblies and hybrid assemblies of short and long
reads.

Conclusions

Automated assembly of long, single-molecule sequencing data reduces the cost
of microbial finishing to $1,000 for most genomes, and future advances in this
technology are expected to drive the cost lower. This is expected to increase
the number of completed genomes, improve the quality of microbial genome
databases, and enable high-fidelity, population-scale studies of pan-genomes
and chromosomal organization.

