---
title: Evaluation of tools for differential gene expression analysis by RNA-seq on
  a 48 biological replicate experiment
categories:
- rnaseq
---
[This paper](http://arxiv.org/ftp/arxiv/papers/1505/1505.02017.pdf) presents a
nice comparison of many RNAseq expression analysis tools.
<!--more-->

> An RNA?seq experiment with 48 biological replicates in each of 2 conditions
was performed to determine the number of biological replicates required, and
to identify the most effective statistical analysis tools for identifying
differential gene expression (DGE). When 3 , seven of the nine tools evaluated
give true positive rates (TPR) of only 20?40%. For high fold?change genes
(|log | 2) the TPR is 85%. Two tools performed poorly; over? or
under?predicting the number of differentially expressed genes. Increasing
replication gives a large increase in TPR when considering all DE genes but
only a small increase for high fold?change genes. Achieving a 85% across all
fold?changes requires 20. For future RNA?seq experiments these results suggest
6, rising to 12 when identifying DGE irrespective of fold?change is important.
For 12 , superior TPR makes edgeR the leading tool tested. For 12, minimizing
false positives is more important and DESeq outperforms the other tools.

