---
title: Why Are Some Genomes Extremely Large?
tags: []
categories:
- blog
---
In our series on [top ten genomes](http://www.homolog.us/blogs/blog/2015/05/08
/top-ten-genomes-x-the-human-genome/), one candidate represented extremely
large genomes. Extremely large genomes are unusual, and it is not unreasonable
to ponder how they ended up being so large. A new paper offers insight
(h/t:Detlef Weigel @plantevolution).
<!--more-->

[Analysis of the giant genomes of Fritillaria (Liliaceae) indicates that a
lack of DNA removal characterizes extreme expansions in genome
size](http://onlinelibrary.wiley.com/doi/10.1111/nph.13471/full)

> Plants exhibit an extraordinary range of genome sizes, varying by >
2000-fold between the smallest and largest recorded values. In the absence of
polyploidy, changes in the amount of repetitive DNA (transposable elements and
tandem repeats) are primarily responsible for genome size differences between
species. However, there is ongoing debate regarding the relative importance of
amplification of repetitive DNA versus its deletion in governing genome size.

Using data from 454 sequencing, we analysed the most repetitive fraction of
some of the largest known genomes for diploid plant species, from members of
Fritillaria.

We revealed that genomic expansion has not resulted from the recent massive
amplification of just a handful of repeat families, as shown in species with
smaller genomes. Instead, the bulk of these immense genomes is composed of
highly heterogeneous, relatively low-abundance repeat-derived DNA, supporting
a scenario where amplified repeats continually accumulate due to infrequent
DNA removal.

Our results indicate that a lack of deletion and low turnover of repetitive
DNA are major contributors to the evolution of extremely large genomes and
show that their size cannot simply be accounted for by the activity of a small
number of high-abundance repeat families.

From the conclusion section -

> Our results from Fritillaria demonstrate that extreme cases of genomic
expansion can take place via the accumulation of highly heterogeneous,
relatively low-abundance, repeat-derived DNA and indicate that a lack of
deletion and low turnover of repetitive DNA play major roles in genome size
evolution. These findings will have important consequences for understanding
the content and evolution of plant genomes. Very large genomes may clearly
still contain highly amplified repeat families that individually have a
substantial impact on genome size, such as is shown here with the high-
abundance tandem repeat in F. affinis (Fig. 2). However, the overall picture
we have revealed, both from analysis of genomic Fritillaria data and from S/L
data from diverse plant species, is not one of genomes growing principally by
the activity of a few repeat families as had previously been suggested.
Whether very large plant genomes (> 20 Gb) exist where significant genome
expansion results solely from the amplification of one or two repeat families
remains to be seen. Irrespective of this, our results, as well as those from
some gymnosperm and animal species, indicate that such a mode of evolution is
not a general feature of extreme genome size expansions. The universality of
the patterns we have revealed awaits testing with data from further species
with giant genomes, such as those found in the Melanthiaceae (Pellicer et al.,
2014) or Viscum (Zonneveld, 2010).

Repetitive DNA can be removed from the genome via homologous and illegitimate
recombination (Fedoroff, 2012); the importance of recombination-based
processes in DNA removal is suggested by the greater estimated rate of DNA
deletion in genomic regions with high recombination rates compared with those
undergoing less recombination (Nam & Ellegren, 2012). A recent theory
presented by Fedoroff (2012) provides a plausible mechanism by which
recombination frequency, and hence DNA removal rate, might be constrained.
Most repetitive elements in plant genomes are highly methylated and contained
within recombinationally inert heterochromatin (Fedoroff, 2012; Henderson,
2012). It is proposed that epigenetic mechanisms, which control the formation
of heterochromatin, evolved to prevent deleterious effects of unconstrained
recombination (Fedoroff, 2012); if unsuppressed, the presence of multiple TE
copies would be expected to stimulate large numbers of ectopic recombination
events (Bennetzen & Wang, 2014). Consequently, efficient epigenetic regulation
of repetitive elements may actually prevent their removal, as they become
locked into tracts of the genome that cannot be accessed by the recombination
machinery (Fedoroff, 2012). If this theory holds true, plant species with
large genomes may accumulate more repetitive DNA because of the rapid action
of epigenetic mechanisms subsequent to amplification, whereas epigenetic
silencing is predicted to reach completion more slowly in species with smaller
genomes, providing a window of opportunity for removal of repetitive DNA via
recombination before heterochromatinization is achieved. This argument runs
counter to the suggestion that epigenetic silencing of repetitive DNA may be
less effective in species with large genomes, thus allowing TEs to proliferate
more easily (Kelly & Leitch, 2011). Although epigenetic mechanisms involved in
regulating activity of repetitive elements have been examined in limited taxa,
there is evidence that they may be less efficient in the larger genome of
Arabidopsis lyrata (1C = 245 Mb; Lysak et al., 2009) than in the smaller
genome of A. thaliana (Hollister et al., 2011). However, initial evidence on
the function of epigenetic mechanisms in F. imperialis indicates that this
species shows all the signatures that are usually associated with strict
epigenetic regulation of repetitive DNA in small genomes (Becher et al.,
2014). Nevertheless, irrespective of whether greater efficiency of epigenetic
control has a role in stimulating genome size expansion, our results provide
clear evidence that a key factor in the evolution of very large genomes is a
lack of DNA removal leading to ongoing accumulation and low turnover of
repetitive and repeat-derived sequences.

Readers may note that [another plant
genome](http://www.homolog.us/blogs/blog/2015/05/08/top-ten-genomes-vii-
bladderwort/) was included in the top ten list for being 'too small', or
rather with hardly any junk DNA. It would be interesting to check what genome
evolution process made that happen.

