---
title: Top Ten Genomes &ndash; (ii) Dynamically Changing Genomes
tags: []
categories:
- blog
---
This week we are talking about highly unusual genomes, and started the
discussion with lamprey [in the previous
commentary](http://www.homolog.us/blogs/blog/2015/05/04/top-ten-
genomes-i-lamprey/). One interesting property of the lamprey genome, if you
remember, is its ability to dynamically shrink during the phases of
development. The genome programmatically chops off about 20% of it, when
lamprey grows from germ line to more complex multi-cellular organism. This
removal not only affects junk dna, but also deletes coding genes.
<!--more-->

That raises the question - how did lamprey genome evolve to programmatically
get rid of parts of its chromosomes? How many other genomes have the same
ability? The answer to the first question is not known, but on the second one
- plenty of other eukaryotic genomes can dynamically change size. Professor
Laura Katz and collaborators put together two wonderful papers compiling a
large list of several dynamically changing genomes.

![katz_130](http://www.homolog.us/blogs/wp-
content/uploads/2015/05/katz_130.jpg)

[The Dynamic Nature of Eukaryotic
Genomes](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2933061/)

> Analyses of diverse eukaryotes reveal that genomes are dynamic, sometimes
dramatically so. In numerous lineages across the eukaryotic tree of life, DNA
content varies within individuals throughout life cycles and among individuals
within species. Discovery of examples of genome dynamism is accelerating as
genome sequences are completed from diverse eukaryotes. Though much is known
about genomes in animals, fungi, and plants, these lineages represent only 3
of the 60200 lineages of eukaryotes. Here, we discuss diverse genomic
strategies in exemplar eukaryotic lineages, including numerous microbial
eukaryotes, to reveal dramatic variation that challenges established views of
genome evolution. For example, in the life cycle of some members of the
radiolaria, ploidy increases from haploid (N) to approximately 1,000N, whereas
intrapopulation variability of the enteric parasite Entamoeba ranges from 4N
to 40N. Variation has also been found within our own species, with substantial
differences in both gene content and chromosome lengths between individuals.
Data on the dynamic nature of genomes shift the perception of the genome from
being fixed and characteristic of a species (typological) to plastic due to
variation within and between species.

\-------------------------------------

[The Dynamic Nature of Genomes across the Tree of
Life](http://www.ncbi.nlm.nih.gov/pubmed/24500971)

> Genomes are dynamic in lineages across the tree of life. Among bacteria and
archaea, for example, DNA content varies throughout life cycles, and nonbinary
cell division in diverse lineages indicates the need for coordination of the
inheritance of genomes. These observations contrast with the textbook view
that bacterial and archaeal genomes are monoploid (i.e., single copied) and
fixed both within species and throughout an individuals lifetime. Here, we
synthesize information on three aspects of dynamic genomes from exemplars
representing a diverse array of bacterial and archaeal lineages: 1) ploidy
level variation, 2) epigenetic mechanisms, and 3) life cycle variation. For
example, the Euryarchaeota analyzed to date are all polyploid, as is the
bacterium Epulopiscium that contains up to tens of thousands of copies of its
genome and reproduces by viviparity. The bacterium Deinococcus radiodurans and
the archaeon Halobacterium sp. NRC-1 can repair a highly fragmented genome
within a few hours. Moreover, bacterial genera such as Dermocarpella and
Planctomyces reproduce by fission (i.e., generating many cells from one cell)
and budding, highlighting the need for regulation of genome inheritance in
these lineages. Combining these data with our previous work on widespread
genome dynamics among eukaryotes, we hypothesize that dynamic genomes are a
rule rather than the exception across the tree of life. Further, we speculate
that all domains may have the ability to distinguish germline from somatic DNA
and that this ability may have been present the last universal common
ancestor.

![Haag_new_portrait](http://www.homolog.us/blogs/wp-content/uploads/2015/05
/Haag_new_portrait-279x300.jpg)

Their widespread existence changes the current view of genomes being static
entities. In fact, even those genomes known to be static may change size, as
[Eric Haag discovered in case of
worm](http://www.homolog.us/blogs/blog/2014/03/31/nwdb-conference-slides-from-
eric-haag-and-update/). We covered his work last year and here were a few
comments he made on his observations.

> 1\. Its important to understand that I discussed quite a number of different
species in my talk, all of which are the genus Caenorhabditis, but only one of
which was C. elegans.

2\. The deletions I discussed happened in the recent evolutionary history of
C. elegans, C. briggsae, and other less-studied species, all united (we think)
only by having adopted a self-fertile hermaphrodite mode of sex.
Philosophically Im not sure when to say the deletions occurred with respect to
taxonomy (i.e. was the pre-hermaphroditic C. elegans ancestor already C.
elegans or something else? But hopefully you get the point: C. elegans, as
conventionally defined as an extant taxon, is already shrunken.

3\. The deletions shouldnt be characterized as programmed, because that
implies precisely regulated and highly predictable loss, and we dont have
evidence for that. What we do have is evidence for loss that is predictable in
quantity and to a lesser extent by broad chromosomal domain (i.e. more in the
ends, less in the middle). These features are probably due to the interaction
of stochastic mutational events (which may have certain biases) and the
landscape of genetic redundancy at the time the lineage switches to self-
fertility.

The genome shrinkage stuff was part of a set of vignettes related to the

consequences of becoming a self-fertile hermaphrodite. I mentioned genome
shrinkage (the issue above), deregulation of sex-biased gene expression, and
increased vulnerability to harm by mating with obligately outcrossing
relatives. I summarized that half of the talk by suggesting that these
consequences eventually slam the door on returning to an outcrossing mode of
reproduction, and make selfing species more subject to extinction when the
environment changes.

I can also summarize the first part of the talk as being about how
independently evolved hermaphroditic species, C. elegans and C. briggsae, have
both repeated and idiosyncratic aspects to the way they regulate the key trait
that makes that possibleXX sperm production. That was based to a large extent
on work from my lab, but also with major contributions from the groups of
Judith Kimble, Tim Schedl, Ron Ellis, and David Pilgrim (who was also a
meeting co-organizer).

The belief that the genomes are static is widespread. Eric Haag mentioned that
the sequencing centers did not pay attention to him at first, when he pointed
out the possibility of worm genomes dynamically changing size. Therefore, we
pick the entire class of dynamically changing genomes as the second candidate
in the top ten genome list.

