---
title: Approaching Single-Cell Sequencing by Understanding NGS Library Complexity
  and Bias
tags: []
categories:
- sponsored
---
Sponsored by [Swift Biosciences, Ann Arbor, MI](http://www.swiftbiosci.com/).
Click [here](http://www.homolog.us/blogs/wp-
content/uploads/2015/10/HighComplex_NGS_AppNote_HRcrops.pdf) to read a pdf
copy of the article.
<!--more-->

**Abstract**

> Demands are growing on genomics to deliver higher quality sequencing data
from samples with less input quantity. As the number of genomic equivalents
decreases at lower input amounts, library bias and library complexity
increasingly affect data quality. Less biased, more complex libraries result
in more even and complete coverage, resulting in better sequencing efficiency
and reduced costs. This application note describes the sequence coverage
performance and preservation of molecular complexity of next generation
sequencing (NGS) libraries generated from human and microbial genomic DNA
using the Accel-NGS 2S DNA Library Kit for whole-genome sequencing (WGS) on
the Illumina platform. Comparisons to the leading commercially available
methods are also presented. From the lowest input DNA quantity supported for
PCR-free libraries (100 ng for Accel-NGS 2S vs. 1 ?g for the leading kit),
this new technology demonstrated more than 50% higher library complexity.
Coverage of extreme GC-rich regions was characterized using the 1,000 bad
promoters (~79% GC) defined by the Broad Institute. The subsequent sequencing
results demonstrated superior coverage performance of these high GC-content
promoters, where 88% were covered better with Accel-NGS 2S than the market
leading kit. Similarly, sequencing analysis of libraries prepared from AT-rich
and GC-rich bacterial genomes showed excellent coverage distribution.


<div id="report"></div>
<script src="/blogs/assets/js/pdfobject.js"></script>
<script>PDFObject.embed("http://www.homolog.us/blogs/wp-content/uploads/2015/10/HighComplex_NGS_AppNote_HRcrops.pdf", "#report");</script>

