---
title: 'After Metagenomics: Drug Discovery from Microbial Genomes'
tags: []
categories:
- blog
---
This year's Nobel prize in medicine went to [William C. Campbell and Satoshi
Omura for discovering new drug Avermectin from microbial
organisms](http://www.homolog.us/blogs/blog/2015/10/05/nobel-prize-for-
diseases-of-poor-people/). They did their work long before the era of low-cost
sequencing.
<!--more-->

Hopefully, high-throughput sequencing will lead to many more success stories
like theirs. The following two papers are excellent reviews on the status of
the field.

[Small molecules from the human
microbiota](http://www.sciencemag.org/content/349/6246/1254766.short)

> Two developments in distinct fields are converging to create interest in
discovering small molecules from the human microbiome. First, the use of
genomics to guide natural product discovery has led to the unexpected
discovery of numerous biosynthetic gene clusters in genomes of the human
microbiota. Second, the microbiome research community is moving from a focus
on whos there? to what are they doing? with an accompanying emphasis on
understanding microbiota-host interactions at the level of molecular
mechanism. This merger has sparked a concerted hunt for the mediators of
microbe-host and microbe-microbe interactions, including microbiota-derived
small molecules.

ADVANCES

Numerous small molecules are known that are produced by the human microbiota.
The microbiota-derived ribosomally synthesized, posttranslationally modified
peptides (RiPPs) include widely distributed lantibiotics and microcins; these
molecules have narrow-spectrum activity and are presumptive mediators of
interactions among closely related species. Another notable RiPP is
Escherichia coli heat-stable enterotoxin, a guanylate cyclase 2C agonist from
which the recently approved gastrointestinal motility drug linaclotide was
derived. Fewer amino acid metabolites are synthesized by the microbiota, but
they are produced at very high levels that vary widely among individuals
(e.g., indoxyl sulfate at 10 to 200 mg/day). Gut bacterial species convert
common dietary amino acids into distinct end products, such as tryptophan to
indoxyl sulfate, indole propionic acid, and tryptamineindicating that humans
with the same diet but different gut colonists can have widely varying gut
metabolic profiles. Microbially produced oligosaccharides differ from other
natural products because they are cell-associated (i.e., nondiffusible) and
because many more biosynthetic loci exist for them than for other small
molecule classes. Well-characterized examples, such as Bacteroides
polysaccharide A, show that oligosaccharides may not simply play a structural
role or mediate adhesion; rather, they can be involved in highly specific
ligand-receptor interactions that result in immune modulation. Similarly, the
(glyco)lipids ?-galactosylceramide and mycolic acid can play roles in immune
signaling. The most prominent microbiota-derived terpenoids are microbial
conversion products of the cholic acid and chenodeoxycholic acid in host bile.
These secondary bile acids can reach high concentration (mM) in the gut and
vary widely in composition among individuals. Several canonical virulence
factors from pathogens are derived from nonribosomal peptides (NRPs) and
polyketides (PKs), but less is known about NRPs and PKs from the commensal
microbiota. A recent computational effort has identified ~14,000 biosynthetic
gene clusters in sequenced genomes from the human microbiota, 3118 of which
were present in one or more of the 752 metagenomic sequence samples from the
NIH Human Microbiome Project. Nearly all of the gene clusters that were
present in >10% of the samples from the body site of origin are
uncharacterized, highlighting the potential for identifying the molecules they
encode and studying their biological activities.

OUTLOOK

There are two central challenges facing the field. The first is to
distinguish, from among thousands of microbiota-derived molecules, which ones
drive a key phenotype at physiologically relevant concentrations. Second,
which experimental systems are appropriate for testing the activity of an
individual molecule from a complex milieu? Meeting these challenges will
require developing new computational and experimental technologies, including
a capacity to identify biosynthetic genes and predict the structure and target
of their biological activity, and systems in which germ-free mice are
colonized by mock communities that differ only by the presence or absence of a
biosynthetic gene cluster.

Commensal organisms of the human microbiota produce many diverse small
molecules with an equally diverse array of targets that can exacerbate or
modulate immune responses and other physiological functions in the host.
Several act as antibacterials to remove competing organisms, but many other
products have unknown targets and effects on commensals and the host.

[Insights into secondary metabolism from a global analysis of prokaryotic
biosynthetic gene clusters](http://www.ncbi.nlm.nih.gov/pubmed/25036635)

> Although biosynthetic gene clusters (BGCs) have been discovered for hundreds
of bacterial metabolites, our knowledge of their diversity remains limited.
Here, we used a novel algorithm to systematically identify BGCs in the
extensive extant microbial sequencing data. Network analysis of the predicted
BGCs revealed large gene cluster families, the vast majority uncharacterized.
We experimentally characterized the most prominent family, consisting of two
subfamilies of hundreds of BGCs distributed throughout the Proteobacteria;
their products are aryl polyenes, lipids with an aryl head group conjugated to
a polyene tail. We identified a distant relationship to a third subfamily of
aryl polyene BGCs, and together the three subfamilies represent the largest
known family of biosynthetic gene clusters, with more than 1,000 members.
Although these clusters are widely divergent in sequence, their small molecule
products are remarkably conserved, indicating for the first time the important
roles these compounds play in Gram-negative cell biology.

