---
title: Top Ten Genomes - (iv) Methanococcus jannaschii
tags: []
categories:
- blog
---
To celebrate the week of Biology of Genome conference, we are compiling a list
of most interesting genomes. Many of these genomes are being discussed at the
conference. Our previous three entries can be seen at the links below.
<!--more-->

[Top Ten Genomes (i) Lamprey](http://www.homolog.us/blogs/blog/2015/05/04/top-
ten-genomes-i-lamprey/)

[Top Ten Genomes (ii) Dynamically Changing
Genomes](http://www.homolog.us/blogs/blog/2015/05/05/top-ten-genomes-ii-
dynamically-changing-genomes/)

[Top Ten Genomes (iii) Paris
japonica](http://www.homolog.us/blogs/blog/2015/05/05/top-ten-genomes-iii-
paris-japonica/)

![carl_woese](http://www.homolog.us/blogs/wp-
content/uploads/2015/05/carl_woese.gif)

![220px-Craigventer2](http://www.homolog.us/blogs/wp-content/uploads/2015/05
/220px-Craigventer2-200x300.jpg)

In 1996, a team led by J. Craig Venter and Carl Woese published the genome of
the first archaea and the results surprised the entire biologist community.
Instead of nicely resolving questions about the origin of three kingdom, it
complicated matters further. The majority of genes related to energy
production, cell division, and metabolism in M. jannaschii were found to be
related to bacteria, whereas the informational component (transcription,
translation, and replication) seemed to have similarity with Eukaryotes.

[Complete Genome Sequence of the Methanogenic Archaeon, Methanococcus
jannaschii

> The complete 1.66-megabase pair genome sequence of an autotrophic archaeon,
Methanococcus jannaschii, and its 58- and 16-kilobase pair extrachromosomal
elements have been determined by whole-genome random sequencing. A total of
1738 predicted protein-coding genes were identified; however, only a minority
of these (38 percent) could be assigned a putative cellular role with high
confidence. Although the majority of genes related to energy production, cell
division, and metabolism in M. jannaschii are most similar to those found in
Bacteria, most of the genes involved in transcription, translation, and
replication in M. jannaschii are more similar to those found in Eukaryotes.

[The complete genome sequence of the hyperthermophilic, sulphate-reducing
archaeon Archaeoglobus
fulgidus](http://www.nature.com/nature/journal/v390/n6658/full/390364a0.html)

> Archaeoglobus fulgidus is the first sulphur-metabolizing organism to have
its genome sequence determined. Its genome of 2,178,400 base pairs contains
2,436 open reading frames (ORFs). The information processing systems and the
biosynthetic pathways for essential components (nucleotides, amino acids and
cofactors) have extensive correlation with their counterparts in the archaeon
Methanococcus jannaschii . The genomes of these two Archaea indicate dramatic
differences in the way these organisms sense their environment, perform
regulatory and transport functions, and gain energy. In contrast to M.
jannaschii , A. fulgidus has fewer restrictionmodification systems, and none
of its genes appears to contain inteins. A quarter (651 ORFs) of the A.
fulgidus genome encodes functionally uncharacterized yet conserved proteins,
two-thirds of which are shared with M. jannaschii (428 ORFs). Another quarter
of the genome encodes new proteins indicating substantial archaeal gene
diversity.

Further sequencing did not help. The origin of Eukaryotes remains to be one of
the biggest unresolved mysteries in evolution.

Here is a 2008 paper by T. Martin Embley and collaborators explaining the
issues.

[The archaebacterial origin of
eukaryotes](http://www.pnas.org/content/105/51/20356.abstract)

> The origin of the eukaryotic genetic apparatus is thought to be central to
understanding the evolution of the eukaryotic cell. Disagreement about the
source of the relevant genes has spawned competing hypotheses for the origins
of the eukaryote nuclear lineage. The iconic rooted 3-domains tree of life
shows eukaryotes and archaebacteria as separate groups that share a common
ancestor to the exclusion of eubacteria. By contrast, the eocyte hypothesis
has eukaryotes originating within the archaebacteria and sharing a common
ancestor with a particular group called the Crenarchaeota or eocytes. Here, we
have investigated the relative support for each hypothesis from analysis of 53
genes spanning the 3 domains, including essential components of the eukaryotic
nucleic acid replication, transcription, and translation apparatus. As an
important component of our analysis, we investigated the fit between model and
data with respect to composition. Compositional heterogeneity is a pervasive
problem for reconstruction of ancient relationships, which, if ignored, can
produce an incorrect tree with strong support. To mitigate its effects, we
used phylogenetic models that allow for changing nucleotide or amino acid
compositions over the tree and data. Our analyses favor a topology that
supports the eocyte hypothesis rather than archaebacterial monophyly and the
3-domains tree of life.

Readers will also find Dan Graur's blog post helpful. It is a section from his
book on genome evolution.

[The Phylogeny of Everything, the Origin of Eukaryotes, and the Rules of
Taxonomy: Death to Archaea, Bacteria, and Eucarya! Long live Archaebacteria,
Eubacteria, Eukaryota, and
Prokaryota!](http://judgestarling.tumblr.com/post/79135526488/the-phylogeny-
of-everything-the-origin-of)

> The origin of the eukaryotic cell is one of the hardest and most interesting
puzzles in evolutionary biology (Lake 2007). Any theory attempting to describe
the evolution of eukaryotes must be able to explain the following seven
eukaryotic characteristics: (1) The eukaryotic cell is considerably more
complex than the prokaryotic cell, possessing, among others a nucleus with a
contiguous endoplasmic reticulum, Golgi bodies, flagella with a 9+2 pattern of
microtubule arrangement, and organelles surrounded by double membranes. (2)
Only eukaryotes achieved great size and morphological complexity, whereas
prokaryotes have remained small and have not evolved either morphological
complexity or multicellularity. (3) The protein-coding genes of eukaryotes are
interspersed with introns that need to be removed prior to translation by
spliceosomes. (4) The process of transcription is physically and temporally
separated from the process of translation. (5) The eukaryote genome consists
of components that are archaebacterial and components that are eubacterial.
(6) The distribution of the archaebacterial and eubacterial genomic components
is not random with respect to function. (7) There are no known precursor
structures among prokaryotes from which such attributes could be derived, and
no intermediate cell types known that would point to a gradual evolutionary
change of a prokaryote into a eukaryote. For all intents and purposes, the
eukaryotic cell represents a sudden organizational upgrade or an evolutionary
leap. Moreover, any theory on eukaryote evolution must provide a reason why
the length of time it took for prokaryotes to evolve out of inanimate matter
is so much shorter than the time it took eukaryotes to evolve out of
prokaryotes.

Given the importance of evolutionary origin of eukaryotes, _Methanococcus
jannaschii_ deserves a place among top ten genomes.

