---
title: An opinionated guide to the proper care and feeding of your transcriptome
categories:
- rnaseq
---
> Characterizing transcriptomes in both model and non-model organisms has
resulted in a massive increase in our understanding of biological phenomena.
This boon, largely made possible via high-throughput sequencing, means that
studies of functional, evolutionary and population genomics are now being done
by hundreds or even thousands of labs around the world. For many, these
studies begin with a de novo transcriptome assembly, which is a technically
complicated process involving several discrete steps. Each step may be
accomplished in one of several different ways, using different software
packages, each producing different results. This analytical complexity begs
the question -- Which method(s) are optimal? Using reference and non-reference
based evaluative methods, I propose a set of guidelines that aim to
standardize and facilitate the process of transcriptome assembly. These
recommendations include the generation of between 20 million and 40 million
sequencing reads from single individual where possible, error correction of
reads, gentle quality trimming, assembly filtering using Transrate and/or gene
expression, annotation using dammit, and appropriate reporting. These
recommendations have been extensively benchmarked and applied to publicly
available transcriptomes, resulting in improvements in both content and
contiguity. To facilitate the implementation of the proposed standardized
methods, I have released a set of version controlled open-sourced code, The
Oyster River Protocol for Transcriptome Assembly, available at http://oyster-
river-protocol.rtfd.org/.
<!--more-->

[Link](http://biorxiv.org/content/early/2015/12/30/035642)
