---
title: Peter Visscher Predicts Future, Makes a Fool of Himself
tags: []
categories:
- blog
---
<!--more-->

<iframe width="560" height="315" src="http://www.youtube.com/embed/IqhlQfXUk7w" frameborder="0"> </iframe>

We came across the hilarious paper - "[Human Complex Trait Genetics in the
21st Century](http://www.genetics.org/content/202/2/377)" - written by GWAS
guru Peter M. Visscher. It reminded us of the following Monte Python video
shown above.

Before you read the article, think about various real scientific disciplines
and check whether anyone predicted the major advances. Did Einstein predict
quantum mechanics? He did not even want to believe that 'God played dice'
after quantum equations were derived. Did someone working on bacterial genomes
predict the existence of an adaptive immune system in the form of CRISPR? Did
those discovering the CRISPR adaptive immune system in bacteria (Barrangou,
Horvath) themselves predict that their discoveries would lead to any use in
human genetics? They argued the opposite.

Most of the life-changing scientific discoveries are discontinuous and cannot
be predicted. However, the expectations are quite different in the scientific
world of silly walk, also known as GWAS. In their world, everything goes
according to the plan, and if not, statistical trickery comes to the rescue. A
few years back, GWAS appeared to be in trouble, because none of the variants
had any impact more than 1-2% in explaining heritability ("[missing
heritability non-problem](http://www.homolog.us/blogs/blog/2014/04/24/a-very-
good-discussion-on-missing-heritability-problem/)"). Peter Visscher came to
the rescue and used mathematics to claim -

[Common SNPs explain a large proportion of the heritability for human
height](http://www.nature.com/ng/journal/v42/n7/abs/ng.608.html)

The real prediction about the future of 'human complex trait genetics in the
21st century' was made in 2000 in the following paper. Their sixteen year old
paper reads like it was written this week.

[How many diseases does it take to map a gene with
SNPs?](http://www.nature.com/ng/journal/v26/n2/full/ng1000_151.html)

> "They all talked at once, their voices insistent and contradictory and
impatient, making of unreality a possibility, then a probability, then an
incontrovertible fact, as people will when their desires become words." W.
Faulkner, The Sound and the Fury, 1929

There are more than a few parallels between the California gold rush and
today's frenetic drive towards linkage disequilibrium (LD) mapping based on
single-nucleotide polymorphisms (SNPs). This is fuelled by a faith that the
genetic determinants of complex traits are tractable, and that knowledge of
genetic variation will materially improve the diagnosis, treatment or
prevention of a substantial fraction of cases of the diseases that constitute
the major public health burden of industrialized nations.

We wonder why Visscher and his followers never mention Weiss/Terwilliger paper
given that the first point of Visscher's paper ('Genetic Data Will Not Be
Limiting') echoes what Weiss anticipated 16 years back.

\------------------------------------------

More specific criticisms of Visscher's article -

1\. **Personalized Genetics and Genomics Will Become an Integral Part of
Health Care and Clinical Practice**

Francis Collins predicted in 1998 that little Johnnies would go to doctor's
offices in 2010 and get their entire future told to them through genetic
tests. Visscher repeating the same canard in 2016 and predicting it to happen
over the 21st century sounds like a defeat.

2\. **Modeling Human Complex Traits in Experimental Organisms Will Become
Obsolete**

This is likely, but has nothing to do with GWAS. Instead major discoveries in
developmental biology over the last 30 years leading to building organoids in
vitro would contribute to such an outcome. Japanese scientists 1, NIH 0.

3\. **In Osteo Population Genetics Studies?**

> it will be possible to take bone samples from a number of individuals who
lived 100, 200, ... 10,000 years ago and infer recent natural selection as if
it was in real time by tracking changes in allele frequencies of variants that
are known (from modern day studies) to be associated with complex traits and
fitness. It might even be possible to study G E interaction by performing gene
mapping on ancestral samples, for example on femur lengths (which is a highly
heritable complex trait). Dig up the bodies!

He is not making a prediction, but rather suggesting a terrible way to waste
money. All it would take to make the 'prediction' work is one of his sidekicks
writing a grant and then publishing an over-hyped paper in Nature Genetics.
Suggested catchy title - "Scientists explain why GWAS scientists are so full
of BS by sequencing all available dead bodies under ground".

God bless the Hindus, who have the good sense of cremating the dead and save
humanity from GWAS lunatics.

4\. **Genetic Data Will Not Be Limiting**

Huh. We all know that the real limiting factor is grants, or public money
being given to various GWAS geniuses.

**Visscher:**

> I predict that the tens of millions of single nucleotide variants and the
many copy number variants that currently segregate in the population will be
whittled down to a much more manageable credible set of plausible causal
variants. I am agnostic as to what the size of that set is going to be (ten
thousand? a hundred thousand? one million?).

[Weiss (in a comment section):](http://www.homolog.us/blogs/blog/2014/04/24/a
-very-good-discussion-on-missing-heritability-problem/)

> I agree with what is said here generally, but I dont think it will make the
missing heritability (Mh) problem go away. There are too many people who for
many different reasons believe that more sophisticated or greater sampling, or
more extensive sequencing and analysis, and larger studies, paired with animal
models, will eventually account for Mh. Whether this is a correct belief or as
much a rationale for funding continual increases in study scale, is debatable.

Rare variants reflect one out that is often invoked, and they certainly
require large studies of one sort or another. The question here is whether
enumerating rare variants and demonstrating their causal role (if it can
actually be done) will do much, especially since most rare variants will be
like their more common known ones, and have very small individual effects.

Another strategy is to blame the mH on interactions. Huge studies or very
clever designs may identify such interactions and evaluate their import,
perhaps at least generically if not by enumeration.

So the problem will, I predict, persist. That doesnt mean the claims about how
to find mH are justified.

