---
title: COCACOLA and BIGMAC Followed by AVOCADO  !
tags: []
categories:
- blog
---
[COCACOLA: binning metagenomic contigs using sequence COmposition, read
CoverAge, CO-alignment, and paired-end read
LinkAge](http://arxiv.org/abs/1604.02512)
<!--more-->

> The advent of next-generation sequencing (NGS) technologies enables
researchers to sequence complex microbial communities directly from
environment. Since assembly typically produces only genome fragments, also
known as contigs, instead of entire genome, it is crucial to group them into
operational taxonomic units (OTUs) for further taxonomic profiling and down-
streaming functional analysis. OTU clustering is also referred to as binning.
We present COCACOLA, a general framework automatically bin contigs into OTUs
based upon sequence composition and coverage across multiple samples.

[BIGMAC : Breaking Inaccurate Genomes and Merging Assembled Contigs for long
read metagenomic assembly](http://biorxiv.org/content/early/2016/03/29/045690)

> The problem of de-novo assembly for metagenomes using only long reads is
gaining attention. We study whether post-processing metagenomic assemblies
with the original input long reads can result in quality improvement. Previous
approaches have focused on pre-processing reads and optimizing assemblers.
BIGMAC takes an alternative perspective to focus on the post-processing step.
Using both the assembled contigs and original long reads as input, BIGMAC
first breaks the contigs at potentially mis-assembled locations and
subsequently scaffolds contigs. Our experiments on metagenomes assembled from
long reads show that BIGMAC can improve assembly quality by reducing the
number of mis-assemblies while maintaining/increasing N50 and N75. The
software is available at https://github.com/kakitone/BIGMAC

[AVOCADO: Visualization of Workflow-Derived Data Provenance for Reproducible
Biomedical Research](http://biorxiv.org/content/early/2016/04/01/044164)

I am getting hungry !

