---
title: A New Metagenome Search Program from the LMAT Group
tags: []
categories:
- blog
---
Readers may compare the following method with [Centrifuge
](http://www.homolog.us/blogs/blog/2015/12/07/centrifuge-a-low-ram-
metagenomic-classifier-from-salzberg-group/) and
[Kaiju](http://www.homolog.us/blogs/blog/2015/12/03/kaiju-fast-and-sensitive-
taxonomic-classification-for-metagenomics/).
<!--more-->

[Link](http://biorxiv.org/content/early/2016/01/14/036681)

> Software for rapid, accurate, and comprehensive microbial profiling of
metagenomic sequence data on a desktop will play an important role in large
scale clinical use of metagenomic data. Here we describe LMAT-ML (Livermore
Metagenomics Analysis Toolkit-Marker Library) which can be run with 24 GB of
DRAM memory, an amount available on many clusters, or with 16 GB DRAM plus a
24 GB low cost commodity flash drive (NVRAM), a cost effective alternative for
desktop or laptop users. We compared results from LMAT with five other rapid,
low-memory tools for metagenome analysis for 131 Human Microbiome Project
samples, and assessed discordant calls with BLAST. All the tools except LMAT-
ML reported overly specific or incorrect species and strain resolution of
reads that were in fact much more widely conserved across species, genera, and
even families. Several of the tools misclassified reads from synthetic or
vector sequence as microbial or human reads as viral. We attribute the high
numbers of false positive and false negative calls to a limited reference
database with inadequate representation of known diversity. Our comparisons
with real world samples show that LMAT-ML is the only tool tested that
classifies the majority of reads, and does so with high accuracy.

