---
title: Two Potentially Important Developments on Nanopore
tags: []
categories:
- blog
---
The first paper demonstrates selective sequencing, whereas the second ones
improves accuracy by introducing circularization.
<!--more-->

[Real time selective sequencing using nanopore
technology](http://biorxiv.org/content/early/2016/02/03/038760.full-
text.pdf+html)

> The Oxford Nanopore MinION is a portable real time sequencing device which
functions by sensing the change in current flow through a nanopore as DNA
passes through it. These current values can be streamed in real time from
individual nanopores as DNA molecules traverse them. Furthermore, the
technology enables individual DNA molecules to be rejected on demand by
reversing the voltage across specific channels. In theory, combining these
features enables selection of individual DNA molecules for sequencing from a
pool, an approach called Read Until. Here we apply dynamic time warping to
match short query current traces to references, demonstrating selection of
specific regions of small genomes, individual amplicons from a group of
targets, or normalisation of amplicons in a set. This is the first
demonstration of direct selection of specific DNA molecules in real time
whilst sequencing on any device and enables many novel uses for the MinION.

[INC-Seq: Accurate single molecule reads using nanopore
sequencing](http://biorxiv.org/content/early/2016/01/27/038042)

> Nanopore sequencing provides a rapid, cheap and portable real-time
sequencing platform with the potential to revolutionize genomics. Several
applications, including RNA-seq, haplotype sequencing and 16S sequencing, are
however limited by its relatively high single read error rate (>10%). We
present INC-Seq (Intramolecular-ligated Nanopore Consensus Sequencing) as a
strategy for obtaining long and accurate nanopore reads starting with low
input DNA. Applying INC-Seq for 16S rRNA based bacterial profiling generated
full-length amplicon sequences with median accuracy >97%. INC-Seq reads enable
accurate species-level classification, identification of species at 0.1%
abundance and robust quantification of relative abundances, providing a cheap
and effective approach for pathogen detection and microbiome profiling on the
MinION system.

